<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="description" content="">
    <meta name="author" content="Mark Otto, Jacob Thornton, and Bootstrap contributors">
    <meta name="generator" content="Hugo 0.88.1">
    <title>DeepCoalCount_network</title>

    <link rel="canonical" href="https://getbootstrap.com/docs/5.1/examples/headers/">
    <link rel="canonical" href="https://getbootstrap.com/docs/5.1/examples/footers/">
    <link rel="canonical" href="https://getbootstrap.com/docs/5.1/examples/jumbotron/">
    
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bootstrap@4.5.3/dist/css/bootstrap.min.css" integrity="sha384-TX8t27EcRE3e/ihU7zmQxVncDAy5uIKz4rEkgIXeMed4M0jlfIDPvg6uqKI2xXr2" crossorigin="anonymous">


    <!-- Bootstrap core CSS -->
    <link href="../assets/dist/css/bootstrap.min.css" rel="stylesheet">

    <style>
      .bd-placeholder-img {
        font-size: 1.125rem;
        text-anchor: middle;
        -webkit-user-select: none;
        -moz-user-select: none;
        user-select: none;
      }

      @media (min-width: 768px) {
        .bd-placeholder-img-lg {
          font-size: 3.5rem;
        }
      }
    </style>

    
    <!-- Custom styles for this template --> 
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
    <link href="/css/headers.css" rel="stylesheet">
    <link href="/css/footers.css" rel="stylesheet">
    <link href="/css/background.css" rel="stylesheet">
    <link href="/css/tables.css" rel="stylesheet">
    <link href="/css/tutorials.css" rel="stylesheet">
   
  </head>

<body class="htmlBody">
    


<main class="mainContainer">
  <script src="/js/nav.js" charset="utf-8"></script>
  <script src="/js/tutorials.js" charset="utf-8"></script>

  <!-- Lightweight client-side loader that feature-detects and load polyfills only when necessary -->
  <script src="https://cdn.jsdelivr.net/npm/@webcomponents/webcomponentsjs@2/webcomponents-loader.min.js"></script>

  <!-- Load the element definition -->
  <script type="module" src="https://cdn.jsdelivr.net/gh/zerodevx/zero-md@1/src/zero-md.min.js"></script>


  <div class="container" >
    <header class="d-flex flex-wrap align-items-center justify-content-center justify-content-md-between py-3 mb-4 border-bottom">

      <div>
              <a href="/" class="d-flex align-items-center col-md-3 mb-2 mb-md-0 text-dark text-decoration-none">
                <img width="250" height="125" src="/media/headerPhoto.png" >
              </a>
      </div>
      <div>
      <ul class="nav col-12 col-md-auto mb-2 justify-content-center mb-md-0">
        <li><a href="/index.html" class="headClick nav-link px-2 link-secondary">Home</a></li>
        <li><a href="/html/about.html" class="headClick nav-link px-2 link-dark">About</a></li>
        <li><a href="/html/software.html" class="headClick nav-link px-2 link-dark">Software</a></li>
        <li><a href="/html/tutorials.html" class="headClick nav-link px-2 link-dark">Tutorials</a></li>
        <li><a href="https://scholar.google.com/citations?hl=en&user=46HLWf8AAAAJ&view_op=list_works&sortby=pubdate" class="headClick nav-link px-2 link-dark">Publications</a></li>
        <li><a href="/html/people.html" class="headClick nav-link px-2 link-dark">People</a></li>
        
        <li><a href="/html/FAQ.html" class="headClick nav-link px-2 link-dark">FAQs</a></li>
      </ul>
      </div>

      <div class="col-md-3 text-end">
      </div>

    </header>
  </div>

  
  <div class="container">
    <!-- Do not set the `src` attribute -->
    <zero-md>
            <!-- Write your markdown inside a `<script type="text/markdown">` tag -->
            <script type="text/markdown">

# DeepCoalCount_network

## Description

Counts the minimum number of extra lineages required to reconcile a list of gene 
trees within the branches of a phylogenetic network. The species network and gene 
trees must be specified in the Rich Newick Format.

The input gene trees can be gene tree distributions inferred from Bayesian methods like MrBayes. 

## Usage

```
DeepCoalCount_network network_ident (gene_tree_ident1 [, gene_tree_ident2...]) [-a taxa map] 
[-b threshold] [-m ac|mul] [resultOutputFile]
```

| Settings                                   |                                                                                                                              |           |
|--------------------------------------------|------------------------------------------------------------------------------------------------------------------------------|-----------|
| *network_ident*                            | The name of the network.                                                                                                     | mandatory |
| *gene_tree_ident1 [, gene_tree_ident2...]* | Comma delimited list of gene tree identifiers. See details.                                                                  | mandatory |
| -m ac or mul                               | Specify the algorithm used for computation (see reference). The default value is ac.                                         | optional  |
| -a taxa map                                | Gene tree / species tree taxa association.                                                                                   | optional  |
| -b threshold                               | Specifies gene trees bootstrap threshold. Edges in the gene trees that have support lower than threshold will be contracted. | optional  |
| resultOutputFile                           | Optional file destination for command output.                                                                                | optional  |

  


By default, it is assumed that only one individual is sampled per species in gene trees. 
However, the option [-a taxa map] allows multiple alleles to be sampled.
  

The -m option is used to specify the algorithm for computation, where mul stands 
for the algorithm based on MUL-trees (Yu Et. Al, under review) and ac stands for 
the algorithm based on ancestral configurations (Yu and Nakhleh, under review). 
They produce exactly the same result, but the latter one is more efficiently in general cases.


## Example


```
#NEXUS
 
BEGIN NETWORKS;
 
Network net = ((A:2,((B:1,C:1):1)X#H1:0::0.3):1,(D:2,X#H1:0::0.7):1);
 
END;
 
 
BEGIN TREES;
 
Tree geneTree1 = (C,((B,D),A));
Tree geneTree2 = (B,(D,(C,A)));
Tree geneTree3 = (D,(B,(C,A)));
 
END;
 
 
BEGIN PHYLONET;
 
DeepCoalCount_network net (geneTree1,geneTree2,geneTree3);
 
END;
```
  


## Command Refernces
* Y. Yu, R.M. Barnett, and L. Nakhleh. Parsimonious inference of hybridization in the presence of incomplete lineage sorting. Under review.
* Y. Yu and L. Nakhleh. Fast algorithms for reconciliation under hybridization and incomplete lineage sorting. Under review.
      </script>
    </zero-md>
</div>
        
                                  
              
    

    <div class="container" height="150px" style="bottom:0;" >
        <footer class="d-flex flex-wrap justify-content-between align-items-center py-3 my-4 border-top">
          <div class="col-md-4 d-flex align-items-center" width="400px">
            <a href="/" class="mb-3 me-2 mb-md-0 text-muted text-decoration-none lh-1">
                <img width="75" height="37.5" src="/media/headerPhoto.png" >
            </a>
            <span class="text-muted" width="350px">&copy; 2022 Luay Nakhleh Lab Group</span>
          </div>
      
          <ul class="nav col-md-4 justify-content-end list-unstyled d-flex">
            <li class="ms-3"><a class="text-muted" href="https://github.com/NakhlehLab"><img width="35" height="35" src="/media/gitLogo.png"></a></li>
            <li class="ms-3"><a class="text-muted" href="https://twitter.com/ricecompsci"><img width="35" height="35" src="/media/twitterLogo.png"></a></li>
          </ul>
        </footer>
     </div>
  
</main>




    <script src="../assets/dist/js/bootstrap.bundle.min.js"></script>

      
  </body>
  
</html>