<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="description" content="">
    <meta name="author" content="Mark Otto, Jacob Thornton, and Bootstrap contributors">
    <meta name="generator" content="Hugo 0.88.1">
    <title>ExploreHypothesis_GibbsSampling Tutorial</title>

    <link rel="canonical" href="https://getbootstrap.com/docs/5.1/examples/headers/">
    <link rel="canonical" href="https://getbootstrap.com/docs/5.1/examples/footers/">
    <link rel="canonical" href="https://getbootstrap.com/docs/5.1/examples/jumbotron/">
    
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bootstrap@4.5.3/dist/css/bootstrap.min.css" integrity="sha384-TX8t27EcRE3e/ihU7zmQxVncDAy5uIKz4rEkgIXeMed4M0jlfIDPvg6uqKI2xXr2" crossorigin="anonymous">


    <!-- Bootstrap core CSS -->
    <link href="../assets/dist/css/bootstrap.min.css" rel="stylesheet">

    <style>
      .bd-placeholder-img {
        font-size: 1.125rem;
        text-anchor: middle;
        -webkit-user-select: none;
        -moz-user-select: none;
        user-select: none;
      }

      @media (min-width: 768px) {
        .bd-placeholder-img-lg {
          font-size: 3.5rem;
        }
      }
    </style>

    
    <!-- Custom styles for this template --> 
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
    <link href="/css/headers.css" rel="stylesheet">
    <link href="/css/footers.css" rel="stylesheet">
    <link href="/css/background.css" rel="stylesheet">
    <link href="/css/tables.css" rel="stylesheet">
    <link href="/css/tutorials.css" rel="stylesheet">
   
  </head>

<body class="htmlBody">
    


<main class="mainContainer">
  <script src="/js/nav.js" charset="utf-8"></script>
  <script src="/js/tutorials.js" charset="utf-8"></script>

  <!-- Lightweight client-side loader that feature-detects and load polyfills only when necessary -->
  <script src="https://cdn.jsdelivr.net/npm/@webcomponents/webcomponentsjs@2/webcomponents-loader.min.js"></script>

  <!-- Load the element definition -->
  <script type="module" src="https://cdn.jsdelivr.net/gh/zerodevx/zero-md@1/src/zero-md.min.js"></script>


  <div class="container" >
    <header class="d-flex flex-wrap align-items-center justify-content-center justify-content-md-between py-3 mb-4 border-bottom">

      <div>
              <a href="/" class="d-flex align-items-center col-md-3 mb-2 mb-md-0 text-dark text-decoration-none">
                <img width="250" height="125" src="/media/headerPhoto.png" >
              </a>
      </div>
      <div>
      <ul class="nav col-12 col-md-auto mb-2 justify-content-center mb-md-0">
        <li><a href="/index.html" class="headClick nav-link px-2 link-secondary">Home</a></li>
        <li><a href="/html/about.html" class="headClick nav-link px-2 link-dark">About</a></li>
        <li><a href="/html/software.html" class="headClick nav-link px-2 link-dark">Software</a></li>
        <li><a href="/html/tutorials.html" class="headClick nav-link px-2 link-dark">Tutorials</a></li>
        <li><a href="https://scholar.google.com/citations?hl=en&user=46HLWf8AAAAJ&view_op=list_works&sortby=pubdate" class="headClick nav-link px-2 link-dark">Publications</a></li>
        <li><a href="/html/people.html" class="headClick nav-link px-2 link-dark">People</a></li>
        
        <li><a href="/html/FAQ.html" class="headClick nav-link px-2 link-dark">FAQs</a></li>
      </ul>
      </div>

      <div class="col-md-3 text-end">
      </div>

    </header>
  </div>

  
  <div class="container">
    <!-- Do not set the `src` attribute -->
    <zero-md>
            <!-- Write your markdown inside a `<script type="text/markdown">` tag -->
            <script type="text/markdown">

# ExploreHypothesis_GibbsSampling

## Description

Using Gibbs Sampling to test the hypothesis given a collection of gene trees.

The network must be specified in the Rich Newick Format. The gene trees must be rooted.

The input gene trees can be gene tree distributions inferred from Bayesian methods like MrBayes.

## Usage

```
ExploreHypothesis_GibbsSampling network_ident geneTreeList [-a taxaMap] [-cl chainLength] [-bl burnInLength] 
[-sf sampleFrequency] [-pt pruningThreshold] [-mb maBranchLength] [-psd] [-pl numberOfThreads] [-di] [resultOutputFile]

```

| Settings             |                                                                                                                               |           |
|----------------------|-------------------------------------------------------------------------------------------------------------------------------|-----------|
| *network_ident*      | The name of the network.                                                                                                      | mandatory |
| *geneTreeList*       | Comma delimited list of gene tree identifiers or comma delimited list of sets of gene tree identifiers.                       | mandatory |
| -a taxaMap           | Gene tree / species tree taxa association.                                                                                    | optional  |
| -cl chainLength      | The length of the Gibbs Sampling. The default value is 11,000.                                                                | optional  |
| -bl burnInLength     | The number of iterations in burn-in period. The default value is 1,000.                                                       | optional  |
| -sf sampleFrequency  | The sample frequency. The default value is 100.                                                                               | optimal   |
| -pt pruningThreshold | The threshold of inheritance probabilities for pruning the reticulation edges in sampled networks. The default value is 0.01. | optional  |
| -mb maxBranchLength  | Maximum branch lengths used in Gibbs Sampling. The default value is 6.                                                        | optional  |
| -psd                 | Using pseudo-likelihood instead of full likelihood.                                                                           | optional  |
| -pl numberOfThreads  | The number of threads used in the computation. The default value is 1.                                                        | optional  |
| -di                  | Output the resulting networks in the format that can be read by Dendroscope .                                                 | optional  |
| resultOutputFile     | Optional file destination for command output.                                                                                 | optional  |
 

The method will output all the samples collected along with their posterior. 
At last, all the sampled networks will be pruned by removing reticulation edges 
whose inheritance probabilities are lower than a threshold. The default value of 
this threshold is 0.01. Users can change it through option -pt. 

By default, it is assumed that only one individual is sampled per species in 
gene trees. However, the option [-a taxaMap] allows multiple alleles to be sampled.

By default, the full likelihood is used in this method. If the dataset is out of 
the scope of full-likelihood computation, users can use pseudo-likelihood instead 
through option -psd.

If users want to run the computation in parallel. Please specify the number of processors through option -pl.

Note that when the program is running, you may see some outputs generated by a
optimization library we are using. In order to ignore these, 
it is recommended to use the optional output file. 


## Example


```
#NEXUS
 
BEGIN NETWORKS;
 
Network net = ((Sbay)#H1,((Smik,(Spar,Scer)),(#H1,Skud)));
 
END;
 
 
BEGIN TREES;
 
Tree gt0  = ((((Scer,Spar),Smik),Skud),Sbay);
Tree gt1  = ((((Scer,Spar),Smik),Skud),Sbay);
Tree gt2  = ((((Scer,Spar),Smik),Skud),Sbay);
Tree gt3  = (((Scer,Spar),(Skud,Sbay)),Smik);
Tree gt4  = (((Scer,Spar),(Smik,Skud)),Sbay);
Tree gt5  = (((Scer,Spar),(Skud,Sbay)),Smik);
Tree gt6  = (((Scer,Spar),(Skud,Sbay)),Smik);
Tree gt7  = (((Scer,Spar),Smik),(Skud,Sbay));
Tree gt8  = (Scer,(Spar,(Smik,(Skud,Sbay))));
Tree gt9  = (((Scer,Spar),(Skud,Sbay)),Smik);
Tree gt10 = (((Scer,Spar),Smik),Skud,Sbay);
Tree gt11 = ((((Scer,Spar),Smik),Skud),Sbay);
Tree gt12 = ((((Scer,Spar),Smik),Skud),Sbay);
Tree gt13 = ((((Scer,Spar),Smik),Skud),Sbay);
Tree gt14 = ((((Scer,Spar),Smik),Skud),Sbay);
Tree gt15 = ((((Scer,Spar),Smik),Skud),Sbay);
Tree gt16 = ((((Scer,Spar),Smik),Skud),Sbay);
Tree gt17 = (((Scer,Spar),Smik),(Skud,Sbay));
Tree gt18 = (((Scer,Spar),Smik),(Skud,Sbay));
Tree gt19 = ((((Scer,Spar),Smik),Skud),Sbay);
Tree gt20 = ((((Scer,Spar),Smik),Skud),Sbay);
Tree gt21 = ((((Scer,Spar),Smik),Skud),Sbay);
Tree gt22 = (((Scer,Spar),Smik),(Skud,Sbay));
Tree gt23 = ((((Scer,Spar),Smik),Skud),Sbay);
Tree gt24 = ((((Scer,Spar),Smik),Skud),Sbay);
Tree gt25 = ((((Scer,Spar),Smik),Skud),Sbay);
Tree gt26 = ((((Scer,Spar),Smik),Sbay),Skud);
Tree gt27 = ((((Scer,Spar),Smik),Skud),Sbay);
Tree gt28 = ((((Scer,Spar),Smik),Skud),Sbay);
Tree gt29 = ((((Scer,Spar),Smik),Skud),Sbay);
Tree gt30 = ((Scer,(Spar,(Skud,Sbay))),Smik);
Tree gt31 = (((Scer,Spar),(Skud,Sbay)),Smik);
Tree gt32 = (((Scer,Spar),(Skud,Sbay)),Smik);
Tree gt33 = ((((Scer,Spar),Smik),Skud),Sbay);
Tree gt34 = (((Scer,Spar),Smik),(Skud,Sbay));
Tree gt35 = (((Scer,Spar),Smik),(Skud,Sbay));
Tree gt36 = ((((Scer,Spar),Skud),Smik),Sbay);
Tree gt37 = ((((Scer,Spar),Smik),Skud),Sbay);
Tree gt38 = ((((Scer,Spar),Smik),Skud),Sbay);
Tree gt39 = ((((Scer,Spar),Smik),Skud),Sbay);
Tree gt40 = ((((Scer,Spar),Smik),Skud),Sbay);
Tree gt41 = (((Scer,Spar),Smik),(Skud,Sbay));
Tree gt42 = ((((Scer,Spar),Smik),Skud),Sbay);
Tree gt43 = (((Scer,Spar),Smik),(Skud,Sbay));
Tree gt44 = ((Scer,Spar),(Smik,(Skud,Sbay)));
Tree gt45 = ((((Scer,Spar),Smik),Sbay),Skud);
Tree gt46 = (((Scer,Spar),Smik),(Skud,Sbay));
Tree gt47 = (((Scer,Spar),(Skud,Sbay)),Smik);
Tree gt48 = ((((Scer,Spar),Smik),Skud),Sbay);
Tree gt49 = ((((Scer,Spar),Smik),Skud),Sbay);
Tree gt50 = ((((Scer,Spar),Smik),Skud),Sbay);
Tree gt51 = ((((Scer,Spar),Smik),Skud),Sbay);
Tree gt52 = ((((Scer,Spar),Smik),Skud),Sbay);
Tree gt53 = ((((Scer,Spar),Smik),Skud),Sbay);
Tree gt54 = ((((Scer,Spar),Smik),Skud),Sbay);
Tree gt55 = (((Scer,Spar),Smik),(Skud,Sbay));
Tree gt56 = ((((Scer,Spar),Smik),Skud),Sbay);
Tree gt57 = ((((Scer,Spar),Smik),Skud),Sbay);
Tree gt58 = (((Scer,Spar),(Skud,Sbay)),Smik);
Tree gt59 = ((((Scer,Spar),Smik),Skud),Sbay);
Tree gt60 = ((((Scer,Spar),Smik),Skud),Sbay);
Tree gt61 = ((((Scer,Spar),Smik),Skud),Sbay);
Tree gt62 = ((Scer,Spar),(Smik,(Skud,Sbay)));
Tree gt63 = ((((Scer,Spar),Smik),Skud),Sbay);
Tree gt64 = ((((Scer,Spar),Smik),Skud),Sbay);
Tree gt65 = ((((Scer,Spar),Smik),Skud),Sbay);
Tree gt66 = (((Scer,Spar),(Smik,Skud)),Sbay);
Tree gt67 = ((((Scer,Spar),Smik),Skud),Sbay);
Tree gt68 = ((((Scer,Spar),Smik),Skud),Sbay);
Tree gt69 = (((Scer,Spar),Smik),(Skud,Sbay));
Tree gt70 = ((((Scer,Spar),Smik),Skud),Sbay);
Tree gt71 = (((Scer,Spar),Smik),(Skud,Sbay));
Tree gt72 = ((((Scer,Spar),Smik),Skud),Sbay);
Tree gt73 = (((Scer,Spar),Smik),(Skud,Sbay));
Tree gt74 = ((((Scer,Spar),Smik),Skud),Sbay);
Tree gt75 = ((((Scer,Spar),Smik),Skud),Sbay);
Tree gt76 = ((((Scer,Spar),Smik),Skud),Sbay);
Tree gt77 = ((((Scer,Spar),Smik),Skud),Sbay);
Tree gt78 = ((((Scer,Spar),Smik),Skud),Sbay);
Tree gt79 = (((Scer,Spar),(Skud,Sbay)),Smik);
Tree gt80 = ((((Scer,Spar),Smik),Skud),Sbay);
Tree gt81 = ((((Scer,Spar),Smik),Skud),Sbay);
Tree gt82 = (((Scer,Spar),(Skud,Sbay)),Smik);
Tree gt83 = ((((Scer,Spar),Smik),Skud),Sbay);
Tree gt84 = ((((Scer,Spar),Smik),Skud),Sbay);
Tree gt85 = ((((Scer,Spar),Smik),Skud),Sbay);
Tree gt86 = ((((Scer,Spar),Smik),Skud),Sbay);
Tree gt87 = ((((Scer,Spar),Smik),Skud),Sbay);
Tree gt88 = ((((Scer,Spar),Smik),Skud),Sbay);
Tree gt89 = ((((Scer,Spar),Smik),Skud),Sbay);
Tree gt90 = ((Scer,Spar),Smik,(Skud,Sbay));
Tree gt91 = (((Scer,Spar),Smik),Skud,Sbay);
Tree gt92 = (Scer,Spar,Smik,Skud,Sbay);
Tree gt93 = (((Scer,Spar),Smik,Skud),Sbay);
Tree gt94 = (Scer,Spar,Smik,(Skud,Sbay));
Tree gt95 = (((Scer,Spar),Skud,Sbay),Smik);
Tree gt96 = ((Scer,Spar),Smik,(Skud,Sbay));
Tree gt97 = (Scer,Spar,(Smik,(Skud,Sbay)));
Tree gt98 = (((Scer,Spar),Smik,Skud),Sbay);
Tree gt99 = ((Scer,Spar),Smik,(Skud,Sbay));
Tree gt100 = (Scer,Spar,Smik,(Skud,Sbay));
Tree gt101 = ((Scer,Spar),Smik,(Skud,Sbay));
Tree gt102 = ((Scer,Spar),Smik,Skud,Sbay);
Tree gt103 = ((Scer,Spar),Smik,Skud,Sbay);
Tree gt104 = (((Scer,Spar),Smik),Skud,Sbay);
Tree gt105 = ((Scer,Spar),Smik,Skud,Sbay);
 
END;
 
 
BEGIN PHYLONET;
 
ExploreHypothesis_GibbsSampling sn (gt0-gt105) gibbsSamplingOutput;
 
END;
```
  


## Command Refernces
* Y. Yu, C. Jermaine, and L. Nakhleh. Exploring Phylogenetic Hypotheses via Gibbs Sampling on Evolutionary Networks. Under Review. 2016.
      </script>
    </zero-md>
</div>
        
                                  
              
    

    <div class="container" height="150px" style="bottom:0;" >
        <footer class="d-flex flex-wrap justify-content-between align-items-center py-3 my-4 border-top">
          <div class="col-md-4 d-flex align-items-center" width="400px">
            <a href="/" class="mb-3 me-2 mb-md-0 text-muted text-decoration-none lh-1">
                <img width="75" height="37.5" src="/media/headerPhoto.png" >
            </a>
            <span class="text-muted" width="350px">&copy; 2022 Luay Nakhleh Lab Group</span>
          </div>
      
          <ul class="nav col-md-4 justify-content-end list-unstyled d-flex">
            <li class="ms-3"><a class="text-muted" href="https://github.com/NakhlehLab"><img width="35" height="35" src="/media/gitLogo.png"></a></li>
            <li class="ms-3"><a class="text-muted" href="https://twitter.com/ricecompsci"><img width="35" height="35" src="/media/twitterLogo.png"></a></li>
          </ul>
        </footer>
     </div>
  
</main>




    <script src="../assets/dist/js/bootstrap.bundle.min.js"></script>

      
  </body>
  
</html>