<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="description" content="">
    <meta name="author" content="Mark Otto, Jacob Thornton, and Bootstrap contributors">
    <meta name="generator" content="Hugo 0.88.1">
    <title>HmmCommand</title>

    <link rel="canonical" href="https://getbootstrap.com/docs/5.1/examples/headers/">
    <link rel="canonical" href="https://getbootstrap.com/docs/5.1/examples/footers/">
    <link rel="canonical" href="https://getbootstrap.com/docs/5.1/examples/jumbotron/">
    
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bootstrap@4.5.3/dist/css/bootstrap.min.css" integrity="sha384-TX8t27EcRE3e/ihU7zmQxVncDAy5uIKz4rEkgIXeMed4M0jlfIDPvg6uqKI2xXr2" crossorigin="anonymous">


    <!-- Bootstrap core CSS -->
    <link href="../assets/dist/css/bootstrap.min.css" rel="stylesheet">

    <style>
      .bd-placeholder-img {
        font-size: 1.125rem;
        text-anchor: middle;
        -webkit-user-select: none;
        -moz-user-select: none;
        user-select: none;
      }

      @media (min-width: 768px) {
        .bd-placeholder-img-lg {
          font-size: 3.5rem;
        }
      }
    </style>

    
    <!-- Custom styles for this template --> 
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
    <link href="/css/headers.css" rel="stylesheet">
    <link href="/css/footers.css" rel="stylesheet">
    <link href="/css/background.css" rel="stylesheet">
    <link href="/css/tables.css" rel="stylesheet">
    <link href="/css/tutorials.css" rel="stylesheet">
   
  </head>

<body class="htmlBody">
    


<main class="mainContainer">
  <script src="/js/nav.js" charset="utf-8"></script>
  <script src="/js/tutorials.js" charset="utf-8"></script>

  <!-- Lightweight client-side loader that feature-detects and load polyfills only when necessary -->
  <script src="https://cdn.jsdelivr.net/npm/@webcomponents/webcomponentsjs@2/webcomponents-loader.min.js"></script>

  <!-- Load the element definition -->
  <script type="module" src="https://cdn.jsdelivr.net/gh/zerodevx/zero-md@1/src/zero-md.min.js"></script>


  <div class="container" >
    <header class="d-flex flex-wrap align-items-center justify-content-center justify-content-md-between py-3 mb-4 border-bottom">

      <div>
              <a href="/" class="d-flex align-items-center col-md-3 mb-2 mb-md-0 text-dark text-decoration-none">
                <img width="250" height="125" src="/media/headerPhoto.png" >
              </a>
      </div>
      <div>
      <ul class="nav col-12 col-md-auto mb-2 justify-content-center mb-md-0">
        <li><a href="/index.html" class="headClick nav-link px-2 link-secondary">Home</a></li>
        <li><a href="/html/about.html" class="headClick nav-link px-2 link-dark">About</a></li>
        <li><a href="/html/software.html" class="headClick nav-link px-2 link-dark">Software</a></li>
        <li><a href="/html/tutorials.html" class="headClick nav-link px-2 link-dark">Tutorials</a></li>
        <li><a href="https://scholar.google.com/citations?hl=en&user=46HLWf8AAAAJ&view_op=list_works&sortby=pubdate" class="headClick nav-link px-2 link-dark">Publications</a></li>
        <li><a href="/html/people.html" class="headClick nav-link px-2 link-dark">People</a></li>
        
        <li><a href="/html/FAQ.html" class="headClick nav-link px-2 link-dark">FAQs</a></li>
      </ul>
      </div>

      <div class="col-md-3 text-end">
      </div>

    </header>
  </div>

  
  <div class="container">
    <!-- Do not set the `src` attribute -->
    <zero-md>
            <!-- Write your markdown inside a `<script type="text/markdown">` tag -->
            <script type="text/markdown">

# HmmCommand

## Description

Uses a hidden markov to assign a species tree to each loci in data coming from a species network.
  
Before using this command, you need to get [lib.zip](/media/lib.zip). Assuming it is then unzipped 
to LocalPath where the unzipped folder is located,

* Mac: use -Djava.library.path=LocalPath/lib, when launching PhyloNet from command line.
* Linux: add LocalPath/lib to your LD_LIBRARY_PATH, before launching PhyloNet from command line.
* Windows: please contact us.
  

## Usage

```
HmmCommand species_network [-gtr] [-allelemap taxaMap]  [-iterations numIterations] [-threads numThreads] [-numberofruns numRuns] 
[-noplots] [-outputdirectory outputLocation]
```

| Settings                        |                                                                          |           |
|---------------------------------|--------------------------------------------------------------------------|-----------|
| *species_network*               | The species network topology of the data. Branch lengths are not needed. | mandatory |
| -gtr                            | Changes the mutation model from the default JC to GTR.                   | optional  |
| -allelemap taxaMap              | Gene tree / species network taxa association.                            | optional  |
| -iterations numIterations       | The number of hill climbing iterations to use for each run.              | optional  |
| -threads numThreads             | The number of threads to use when processing.                            | optional  |
| -numberofruns numRuns           | The number of times to restart the hill climbing process.                | optional  |
| -noplots                        | Stop showing plots when finished.                                        | optional  |
| -outputdirectory outputLocation | The directory to store the output of the run.                            | optional  |

## Example


```
#NEXUS
 
BEGIN NETWORKS;
Network net = ((A:3,ANC#H1:2.25):2,((B:.75)ANC#H1:0,C:.75):4.25);
END;
 
Begin DATA;
   dimensions ntax=3 nchar=6;
        format datatype=dna symbols="ACTG" missing=? gap=-;
        matrix
1   AGGTGG
2   AACCTT
3   ACCTGT
;
END;
 
BEGIN PHYLONET;
HmmCommand net -allelemap <A:1; B:2; C:3> -outputdirectory "foo" -threads 4 -iterations 300;
END;

```

## Command References

* Liu KJ, Dai J, Truong K, Song Y, Kohn MH, and Nakhleh L (2014) An HMM-Based Comparative Genomic Framework for Detecting Introgression in Eukaryotes. PLoS Comput Biol 10(6): e1003649.
* Liu KJ, Steinberg E, Yozzo A, Song Y, Kohn MH, and Nakhleh L (2014) Interspecific Introgressive Origin of Genomic Diversity in the House Mouse. Under review. 
  
      </script>
    </zero-md>
</div>
        
                                  
              
    

    <div class="container" height="150px" style="bottom:0;" >
        <footer class="d-flex flex-wrap justify-content-between align-items-center py-3 my-4 border-top">
          <div class="col-md-4 d-flex align-items-center" width="400px">
            <a href="/" class="mb-3 me-2 mb-md-0 text-muted text-decoration-none lh-1">
                <img width="75" height="37.5" src="/media/headerPhoto.png" >
            </a>
            <span class="text-muted" width="350px">&copy; 2022 Luay Nakhleh Lab Group</span>
          </div>
      
          <ul class="nav col-md-4 justify-content-end list-unstyled d-flex">
            <li class="ms-3"><a class="text-muted" href="https://github.com/NakhlehLab"><img width="35" height="35" src="/media/gitLogo.png"></a></li>
            <li class="ms-3"><a class="text-muted" href="https://twitter.com/ricecompsci"><img width="35" height="35" src="/media/twitterLogo.png"></a></li>
          </ul>
        </footer>
     </div>
  
</main>




    <script src="../assets/dist/js/bootstrap.bundle.min.js"></script>

      
  </body>
  
</html>