<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="description" content="">
    <meta name="author" content="Mark Otto, Jacob Thornton, and Bootstrap contributors">
    <meta name="generator" content="Hugo 0.88.1">
    <title>MCMC_BiMarkers Tutorial</title>

    <link rel="canonical" href="https://getbootstrap.com/docs/5.1/examples/headers/">
    <link rel="canonical" href="https://getbootstrap.com/docs/5.1/examples/footers/">
    <link rel="canonical" href="https://getbootstrap.com/docs/5.1/examples/jumbotron/">
    
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bootstrap@4.5.3/dist/css/bootstrap.min.css" integrity="sha384-TX8t27EcRE3e/ihU7zmQxVncDAy5uIKz4rEkgIXeMed4M0jlfIDPvg6uqKI2xXr2" crossorigin="anonymous">


    <!-- Bootstrap core CSS -->
    <link href="../assets/dist/css/bootstrap.min.css" rel="stylesheet">

    <style>
      .bd-placeholder-img {
        font-size: 1.125rem;
        text-anchor: middle;
        -webkit-user-select: none;
        -moz-user-select: none;
        user-select: none;
      }

      @media (min-width: 768px) {
        .bd-placeholder-img-lg {
          font-size: 3.5rem;
        }
      }
    </style>

    
    <!-- Custom styles for this template --> 
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
    <link href="/css/headers.css" rel="stylesheet">
    <link href="/css/footers.css" rel="stylesheet">
    <link href="/css/background.css" rel="stylesheet">
    <link href="/css/tables.css" rel="stylesheet">
    <link href="/css/tutorials.css" rel="stylesheet">
   
  </head>

<body class="htmlBody">
    


<main class="mainContainer">
  <script src="/js/nav.js" charset="utf-8"></script>
  <script src="/js/tutorials.js" charset="utf-8"></script>
  <script src="https://cdn.jsdelivr.net/gh/google/code-prettify@master/loader/run_prettify.js"></script>

  <!-- Lightweight client-side loader that feature-detects and load polyfills only when necessary -->
  <script src="https://cdn.jsdelivr.net/npm/@webcomponents/webcomponentsjs@2/webcomponents-loader.min.js"></script>

  <!-- Load the element definition -->
  <script type="module" src="https://cdn.jsdelivr.net/gh/zerodevx/zero-md@1/src/zero-md.min.js"></script>

  <div class="container" >
    <header class="d-flex flex-wrap align-items-center justify-content-center justify-content-md-between py-3 mb-4 border-bottom">

      <div>
              <a href="/" class="d-flex align-items-center col-md-3 mb-2 mb-md-0 text-dark text-decoration-none">
                <img width="250" height="125" src="/media/headerPhoto.png" >
              </a>
      </div>
      <div>
      <ul class="nav col-12 col-md-auto mb-2 justify-content-center mb-md-0">
        <li><a href="/index.html" class="headClick nav-link px-2 link-secondary">Home</a></li>
        <li><a href="/html/about.html" class="headClick nav-link px-2 link-dark">About</a></li>
        <li><a href="/html/software.html" class="headClick nav-link px-2 link-dark">Software</a></li>
        <li><a href="/html/tutorials.html" class="headClick nav-link px-2 link-dark">Tutorials</a></li>
        <li><a href="https://scholar.google.com/citations?hl=en&user=46HLWf8AAAAJ&view_op=list_works&sortby=pubdate" class="headClick nav-link px-2 link-dark">Publications</a></li>
        <li><a href="/html/people.html" class="headClick nav-link px-2 link-dark">People</a></li>
        
        <li><a href="/html/FAQ.html" class="headClick nav-link px-2 link-dark">FAQs</a></li>
      </ul>
      </div>

      <div class="col-md-3 text-end">
      </div>

    </header>
  </div>

    <div class="container">
        <!-- Do not set the `src` attribute -->
        <zero-md>
                <!-- Write your markdown inside a `<script type="text/markdown">` tag -->
                <script type="text/markdown">

# MCMC_BiMarkers

## Description and Download

This method performs Bayesian estimation of the posterior distribution of phylogenetic networks given bi-allelic genetic markers (SNPs, AFLPs, etc).

This method also uses the Jeigen package, so if you encounter errors like:

>java.lang.NoClassDefFoundError: Could not initialize class jeigen.JeigenJna$Jeigen

You need to follow this instruction to build Jeigen on your machine: https://github.com/hughperkins/jeigen#how-to-build-linux . After you build Jeigen, copy the library file to .jeigen folder in the home path.
  
## Usage

```
MCMC_BiMarkers [-diploid] [-dominant dominantMarker] [-op] [-cl chainLength] [-bl burnInLength] [-sf sampleFrequency] 
[-sd seed] [-pl parallelThreads] [-mc3 temperatureList] [-mr maxReticulation] [-tm taxonMap] [-fixtheta theta] [-varytheta]
[-esptheta] [-pp poissonParameter] [-dd] [-ee expPrior] [-snet startingNetwork] [-ptheta startingThetaPrior] [-pi0 PI0] 
```

| MCMC Settings              |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           |          |
|----------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|----------|
| -cl chainLength            | The length of the MCMC chain. Example: 500,000.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           | optional |
| -bl burnInLength           | The number of iterations in burn-in period. Example: 200,000.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             | optional |
| -sf sampleFrequency        | The sample frequency. The default value is 500.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           | optional |
| -sd seed                   | The random seed. The default seed is 12345678.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            | optional |
| -pl parallelThreads        | The number of threads running in parallel. The default value is the number of threads in your machine.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    | optional |
| **MC3 Settings**               |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           |          |
| -mc3 temperatureList       | The list of temperatures for the Metropolis-coupled MCMC chains. For example, -mc3 (2.0, 3.0)indicates  two hot chains with temperatures 2.0 and 3.0 respectively will be run  along with the cold chain with temperature 1.0. By default only the cold  chain will be run. Note that The temperatures should be DIFFERENT! For example, -mc3 (2.0, 2.0, 3.0) is invalid. The temperature of the cold chain should NOT be included. For example, -mc3 (1.0, 2.0, 3.0) is incorrect. Metropolis-coupled  MCMC leads to faster convergence and better mixing, however, the  running time increases linearly with the number of chains. We suggest  you first run a standard MCMC chain (cold chain) without this command.  If the trace plot indicates the chain is not mixed well (jagged, stuck  in local maxima for a long time), then try this command. | optional |
| **Inference Settings**         |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           |          |
| -mr maxReticulation        | The maximum number of reticulation nodes in the sampled phylogenetic networks. The default value is 4.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    | optional |
| -taxa taxaList             | The taxa used for inference. For example, -taxa (a,b,c)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   | required |
| -tm taxonMap               | Gene tree / species tree taxa association.  By default, it is assumed that only one individual is sampled per  species in gene trees. This option allows multiple alleles to be  sampled. For example, the gene tree is (((a1,a2),(b1,b2)),c); and the species tree is ((a,b),c);, the command is -tm <a:a1,a2; b:b1,b2;c:c>. Note that the taxa association should cover all species, e.g. -tm <a:a1,a2; b:b1,b2> is incorrect because c:c is dropped out.  If  the set of taxa here is different than that in "-taxa taxaList", a  common subset of taxa of both parameters will be used for inference.                                                                                                                                                                                                                                                 | optional |
| -fixtheta theta            | Fix  the population mutation rates associated with all branches of the  phylogenetic network to this given value (theta). By default, we  estimate a constant population size across all branches.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        | optional |
| -varytheta                 | The  population mutation rates across all branches may be different when  estimating them. By default, we estimate a constant population size  across all branches.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       | optional |
| -esptheta                  | Estimate the mean value of prior of population mutation rates.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            | optional |
| **Prior Settings**             |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           |          |
| -pp poissonParam           | The Poisson parameter in the prior on the number of reticulation nodes. The default value is 1.0.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         | optional |
| -dd                        | Disable the prior on the diameters of hybridizations. By default this prior on is exp(10).                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                | optional |
| -ee                        | Enable the Exponential(10) prior on the divergence times of nodes in the phylogenetic network. By default we use Uniform prior.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           | optional |
| **Starting State Settings**    |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           |          |
| -snet                      | Specify  the starting network. The input network should be ultrametric with  divergence times in units of expected number of mutations per site,  inheritance probabilities and population sizes in units of population  mutation rate (optional). See example below. The default starting  network is the MDC trees given starting gene trees.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           | optional |
| -ptheta startingThetaPrior | Specify the mean value of prior of population mutation rate (startingThetaPrior). The default value is 0.036. If -esptheta is used, startingThetaPrior will be treated as the starting value, otherwise startingThetaPrior will be treated as the fixed mean value of prior of population mutation rates.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 | optional |
| **Data related settings**      |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           |          |
| -diploid                   | Specify whether sequence sampled from diploids.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           | optional |
| -dominant dominantMarker   | Specify which marker is dominant if the data is dominant. Either '0' or '1'.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           | optional |
| -op                        | Specify  whether or not to ignore all monomorphic sites. If this option is used,  the data will be treated as containing only polymorphic sites.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          | optional |

## Examples


Download: [MCMCseq_example0.nex](/media/nexus/MCMCseq_example0.txt)
```
#NEXUS
Begin data;
Dimensions ntax=5 nchar=10;
Format datatype=dna symbols="012" missing=? gap=-;
Matrix
  

a 1010101010
b 1100100110
c1 1010100011
c2 1001110100
d 1000011110

;End;
  
BEGIN PHYLONET;
MCMC_BiMarkers -cl 500000 -bl 200000 -sf 500 -diploid -dominant 1 -op -varytheta -pp 2.0 -ee 2.0 -mr 1 -pl 4 -esptheta -ptheta 0.3
-sd 12345678

-taxa (a,b,c1,c2,d)
-tm <A:a; B:b;C:c1,c2;D:d>;

END;
```

Note that an empty line should be left after "Matrix".

This command will run MCMC chain of 500000 iterations with 200000 burn-in iterations, and one sample will be collected every 500 iterations. The taxa are diploids and 1 is the dominant marker. Only polymorphic sites will be used. We will estimate population mutation rates for every branches, and they may be different. A Poisson prior of 2.0 will be adopted, and a Exponential(2.0) prior will be adopted. The number of reticulation nodes is limited to 1. We will sample the mean value of prior of population mutation rates, and the starting value of 0.3 is given. We use the random seed of 12345678. In the end, we indicate the mapping from taxa to species.


## Command References 

  * D.Wen and L. Nakhleh. Co-estimating reticulate phylogenies and gene trees on sequences from multiple independent loci. Submitted. 
  * Gronau, Ilan, et al. Bayesian inference of ancient human demography from individual genome sequences. Nature genetics 43.10 (2011): 1031-1034.
          </script>
        </zero-md>
    </div>
      
    

        
                                  
              
    

    <div class="container" height="150px" style="bottom:0;" >
        <footer class="d-flex flex-wrap justify-content-between align-items-center py-3 my-4 border-top">
          <div class="col-md-4 d-flex align-items-center" width="400px">
            <a href="/" class="mb-3 me-2 mb-md-0 text-muted text-decoration-none lh-1">
                <img width="75" height="37.5" src="/media/headerPhoto.png" >
            </a>
            <span class="text-muted" width="350px">&copy; 2022 Luay Nakhleh Lab Group</span>
          </div>
      
          <ul class="nav col-md-4 justify-content-end list-unstyled d-flex">
            <li class="ms-3"><a class="text-muted" href="https://github.com/NakhlehLab"><img width="35" height="35" src="/media/gitLogo.png"></a></li>
            <li class="ms-3"><a class="text-muted" href="https://twitter.com/ricecompsci"><img width="35" height="35" src="/media/twitterLogo.png"></a></li>
          </ul>
        </footer>
     </div>
  
</main>




    <script src="../assets/dist/js/bootstrap.bundle.min.js"></script>

      
  </body>
  
</html>