<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="description" content="">
    <meta name="author" content="Mark Kessler">
    <meta name="generator" content="Hugo 0.88.1">
    <title>NetMerger</title>

    <link rel="canonical" href="https://getbootstrap.com/docs/5.1/examples/headers/">
    <link rel="canonical" href="https://getbootstrap.com/docs/5.1/examples/footers/">
    <link rel="canonical" href="https://getbootstrap.com/docs/5.1/examples/jumbotron/">
    
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bootstrap@4.5.3/dist/css/bootstrap.min.css" integrity="sha384-TX8t27EcRE3e/ihU7zmQxVncDAy5uIKz4rEkgIXeMed4M0jlfIDPvg6uqKI2xXr2" crossorigin="anonymous">


    <!-- Bootstrap core CSS -->
    <link href="../assets/dist/css/bootstrap.min.css" rel="stylesheet">

    <style>
      .bd-placeholder-img {
        font-size: 1.125rem;
        text-anchor: middle;
        -webkit-user-select: none;
        -moz-user-select: none;
        user-select: none;
      }

      @media (min-width: 768px) {
        .bd-placeholder-img-lg {
          font-size: 3.5rem;
        }
      }
    </style>

    
    <!-- Custom styles for this template --> 
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
    <link href="/css/headers.css" rel="stylesheet">
    <link href="/css/footers.css" rel="stylesheet">
    <link href="/css/background.css" rel="stylesheet">
    <link href="/css/tables.css" rel="stylesheet">
    <link href="/css/tutorials.css" rel="stylesheet">
   
  </head>

<body class="htmlBody">
    


<main class="mainContainer">
  <script src="/js/nav.js" charset="utf-8"></script>
  <script src="/js/tutorials.js" charset="utf-8"></script>

  <!-- Lightweight client-side loader that feature-detects and load polyfills only when necessary -->
  <script src="https://cdn.jsdelivr.net/npm/@webcomponents/webcomponentsjs@2/webcomponents-loader.min.js"></script>

  <!-- Load the element definition -->
  <script type="module" src="https://cdn.jsdelivr.net/gh/zerodevx/zero-md@1/src/zero-md.min.js"></script>


  <div class="container" >
    <header class="d-flex flex-wrap align-items-center justify-content-center justify-content-md-between py-3 mb-4 border-bottom">

      <div>
              <a href="/" class="d-flex align-items-center col-md-3 mb-2 mb-md-0 text-dark text-decoration-none">
                <img width="250" height="125" src="/media/headerPhoto.png" >
              </a>
      </div>
      <div>
      <ul class="nav col-12 col-md-auto mb-2 justify-content-center mb-md-0">
        <li><a href="/index.html" class="headClick nav-link px-2 link-secondary">Home</a></li>
        <li><a href="/html/about.html" class="headClick nav-link px-2 link-dark">About</a></li>
        <li><a href="/html/software.html" class="headClick nav-link px-2 link-dark">Software</a></li>
        <li><a href="/html/tutorials.html" class="headClick nav-link px-2 link-dark">Tutorials</a></li>
        <li><a href="https://scholar.google.com/citations?hl=en&user=46HLWf8AAAAJ&view_op=list_works&sortby=pubdate" class="headClick nav-link px-2 link-dark">Publications</a></li>
        <li><a href="/html/people.html" class="headClick nav-link px-2 link-dark">People</a></li>
        
        <li><a href="/html/FAQ.html" class="headClick nav-link px-2 link-dark">FAQs</a></li>
      </ul>
      </div>

      <div class="col-md-3 text-end">
      </div>

    </header>
  </div>

  
  <div class="container">
    <!-- Do not set the `src` attribute -->
    <zero-md>
            <!-- Write your markdown inside a `<script type="text/markdown">` tag -->
            <script type="text/markdown">

# NetMerger

## Description

A divide-and-conquer approach to infer large network from multi-locus sequence alignments or bi-allelic markers.

## Usage

#### To Reduce the Number of Trinets

This mode will generate a bunch of triplets using the heuristic to reduce the 
number of subproblems, by covering the internal branches of gene trees.

```
NetMerger -mode "Triplets" [-outgroup outgroup] [-tm taxonMap] [-gts gtsFilePath] [-triplets tripletsFilePath]
```

| Triplets Settings          |                                                                                                                |          |
|----------------------------|----------------------------------------------------------------------------------------------------------------|----------|
| -outgroup outgroup         | The taxon in species network designated to be the "outgroup".                                                  | required |
| -tm taxonMap               | Gene tree / species tree taxa association of the full data set.                                                | required |
| -gts gtsFilePath           | The path to the text file, in which each line is a gene tree.                                                  | required |
| -triplets tripletsFilePath | The path to the file to be written with triplets. Each line of the file has three species, separated by space. | required |
  

#### To Divide the Data Set into Subsets

 
In this mode, user can provide a "template" nexus file with regular "MCMC_SEQ"  
or "MCMC_BiMarkers" command. Then this mode will generate a bunch of new nexus 
files, under the same folder as the template, and replace the taxon mapping 
in the template file by a subset of that mapping. Therefore if PhyloNet uses 
a new nexus file as an input, a trinet will be inferred. The user can specify 
whether the full set or reduced set of trinets needs to be inferred.

```
NetMerger -mode "Nex" [-nex nexFilePath] [-tm taxonMap] [-triplets tripletsFilePath] 
```

| Nex Settings               |                                                                                                                                                                           |          |
|----------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------|----------|
| -nex nexFilePath           | The path to the "template" nexus file.                                                                                                                                    | required |
| -tm taxonMap               | Gene tree / species tree taxa association of the full data set.                                                                                                           | required |
| -triplets tripletsFilePath | The  path to the text file with triplets. Each line of the file has three  species, separated by space. If not specified, the full set of  subproblems will be generated. | optional |
 

#### To Merge Trinets

Users can run PhyloNet with generated new nexus files. After trinets are inferred, 
those results generated by MCMC_SEQ or MCMC_BiMarkers can be merged by NetMerger. 
If the set of trinets needs to be enriched, a list of triplets will be provided, 
otherwise a final network is produced.

```
NetMerger -mode "Result" [-cl chainLength] [-bl burnInLength] [-sf sampleFrequency] 
[-outgroup outgroup] [-inputFolder inputFolder]
```


| Result Settings           |                                                                                                                                |          |
|---------------------------|--------------------------------------------------------------------------------------------------------------------------------|----------|
| -cl chainLength           | The length of the MCMC chain.                                                                                                  | required |
| -bl burnInLength          | The number of iterations in burn-in period of MCMC inference.                                                                  | required |
| -sf sampleFrequency       | The sample frequency in MCMC inference.                                                                                        | required |
| -outgroup outgroup        | The taxon in species network designated to be the "outgroup".                                                                  | required |
| -inputFolder inputFolder  | The folder where results of MCMC_SEQ or MCMC_BiMarkers locate. Each file ends in ".out" is a single inference of a subnetwork. | required |




## Examples


```
#NEXUS 

BEGIN PHYLONET; 

NetMerger -mode "Triplets" -gts "/Users/zhujiafan/Documents/BioinfoData/SuperNetwork/NetMerger/gts.txt"

 -triplets "/Users/zhujiafan/Documents/BioinfoData/SuperNetwork/NetMerger/triplets.txt" 

 -outgroup "Z"

 -tm <A:A_0,A_1;B:B_0,B_1;C:C_0,C_1;D:D_0,D_1;E:E_0,E_1;F:F_0,F_1;G:G_0,G_1;H:H_0,H_1;I:I_0,I_1;J:J_0,J_1;K:K_0,K_1;L:L_0,L_1;M:M_0,M_1;N:N_0,N_1;O:O_0,O_1;P:P_0,P_1;Z:Z_0,Z_1> ;

END;
```

The above command writes triplets to file "/Users/zhujiafan/Documents/BioinfoData/SuperNetwork/NetMerger/triplets.txt" 
according to the gene trees in "/Users/zhujiafan/Documents/BioinfoData/SuperNetwork/NetMerger/gts.txt".

```
#NEXUS 

BEGIN PHYLONET; 

NetMerger -mode "Nex" -nex "/Users/zhujiafan/Documents/BioinfoData/SuperNetwork/NetMerger/newtest.nex"

 -triplets "/Users/zhujiafan/Documents/BioinfoData/SuperNetwork/NetMerger/triplets.txt"

 -tm <A:A_0,A_1;B:B_0,B_1;C:C_0,C_1;D:D_0,D_1;E:E_0,E_1;F:F_0,F_1;G:G_0,G_1;H:H_0,H_1;I:I_0,I_1;J:J_0,J_1;K:K_0,K_1;L:L_0,L_1;M:M_0,M_1;N:N_0,N_1;O:O_0,O_1;P:P_0,P_1;Z:Z_0,Z_1> ;

END;
```

The above command produces "newtest_0.nex", "newtest_1.nex", etc, under the folder 
"/Users/zhujiafan/Documents/BioinfoData/SuperNetwork/NetMerger/". The number of new
nexus files equals to the number of triplets provided in 
"/Users/zhujiafan/Documents/BioinfoData/SuperNetwork/NetMerger/triplets.txt". 
Each new nexus file can be used as the input of PhyloNet.


```
#NEXUS

BEGIN PHYLONET;

NetMerger -cl 6000000 -bl 3000000 -sf 5000 -inputFolder "/Users/zhujiafan/Documents/BioinfoData/results/"
-outgroup "Z" ;
END;

```

The above command parses all files end with ".out" under the folder "/Users/zhujiafan/Documents/BioinfoData/results/", 
then merges those trinets generated by Bayesian inference. The command terminates and 
provides additional triplets if the set of trinets needs to be enriched, 
otherwise it produces the final network.


## Command References

* D.Wen and L. Nakhleh. Co-estimating reticulate phylogenies and gene trees on sequences from multiple independent loci. 
* J. Zhu, X. Liu, H.A. Ogilvie and L. Nakhleh. A Divide-and-Conquer Method for Scalable Phylogenetic Network Inference from Multi-locus Data

      </script>
    </zero-md>
</div>
        
                                  
              
    

    <div class="container" height="150px" style="bottom:0;" >
        <footer class="d-flex flex-wrap justify-content-between align-items-center py-3 my-4 border-top">
          <div class="col-md-4 d-flex align-items-center" width="400px">
            <a href="/" class="mb-3 me-2 mb-md-0 text-muted text-decoration-none lh-1">
                <img width="75" height="37.5" src="/media/headerPhoto.png" >
            </a>
            <span class="text-muted" width="350px">&copy; 2022 Luay Nakhleh Lab Group</span>
          </div>
      
          <ul class="nav col-md-4 justify-content-end list-unstyled d-flex">
            <li class="ms-3"><a class="text-muted" href="https://github.com/NakhlehLab"><img width="35" height="35" src="/media/gitLogo.png"></a></li>
            <li class="ms-3"><a class="text-muted" href="https://twitter.com/ricecompsci"><img width="35" height="35" src="/media/twitterLogo.png"></a></li>
          </ul>
        </footer>
     </div>
  
</main>




    <script src="../assets/dist/js/bootstrap.bundle.min.js"></script>

      
  </body>
  
</html>