<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="description" content="">
    <meta name="author" content="Mark Otto, Jacob Thornton, and Bootstrap contributors">
    <meta name="generator" content="Hugo 0.88.1">
    <title>ProcessGT</title>

    <link rel="canonical" href="https://getbootstrap.com/docs/5.1/examples/headers/">
    <link rel="canonical" href="https://getbootstrap.com/docs/5.1/examples/footers/">
    <link rel="canonical" href="https://getbootstrap.com/docs/5.1/examples/jumbotron/">
    
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bootstrap@4.5.3/dist/css/bootstrap.min.css" integrity="sha384-TX8t27EcRE3e/ihU7zmQxVncDAy5uIKz4rEkgIXeMed4M0jlfIDPvg6uqKI2xXr2" crossorigin="anonymous">


    <!-- Bootstrap core CSS -->
    <link href="../assets/dist/css/bootstrap.min.css" rel="stylesheet">

    <style>
      .bd-placeholder-img {
        font-size: 1.125rem;
        text-anchor: middle;
        -webkit-user-select: none;
        -moz-user-select: none;
        user-select: none;
      }

      @media (min-width: 768px) {
        .bd-placeholder-img-lg {
          font-size: 3.5rem;
        }
      }
    </style>

    
    <!-- Custom styles for this template --> 
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
    <link href="/css/headers.css" rel="stylesheet">
    <link href="/css/footers.css" rel="stylesheet">
    <link href="/css/background.css" rel="stylesheet">
    <link href="/css/tables.css" rel="stylesheet">
    <link href="/css/tutorials.css" rel="stylesheet">
   
  </head>

<body class="htmlBody">
    


<main class="mainContainer">
  <script src="/js/nav.js" charset="utf-8"></script>
  <script src="/js/tutorials.js" charset="utf-8"></script>

  <!-- Lightweight client-side loader that feature-detects and load polyfills only when necessary -->
  <script src="https://cdn.jsdelivr.net/npm/@webcomponents/webcomponentsjs@2/webcomponents-loader.min.js"></script>

  <!-- Load the element definition -->
  <script type="module" src="https://cdn.jsdelivr.net/gh/zerodevx/zero-md@1/src/zero-md.min.js"></script>


  <div class="container" >
    <header class="d-flex flex-wrap align-items-center justify-content-center justify-content-md-between py-3 mb-4 border-bottom">

      <div>
              <a href="/" class="d-flex align-items-center col-md-3 mb-2 mb-md-0 text-dark text-decoration-none">
                <img width="250" height="125" src="/media/headerPhoto.png" >
              </a>
      </div>
      <div>
      <ul class="nav col-12 col-md-auto mb-2 justify-content-center mb-md-0">
        <li><a href="/index.html" class="headClick nav-link px-2 link-secondary">Home</a></li>
        <li><a href="/html/about.html" class="headClick nav-link px-2 link-dark">About</a></li>
        <li><a href="/html/software.html" class="headClick nav-link px-2 link-dark">Software</a></li>
        <li><a href="/html/tutorials.html" class="headClick nav-link px-2 link-dark">Tutorials</a></li>
        <li><a href="https://scholar.google.com/citations?hl=en&user=46HLWf8AAAAJ&view_op=list_works&sortby=pubdate" class="headClick nav-link px-2 link-dark">Publications</a></li>
        <li><a href="/html/people.html" class="headClick nav-link px-2 link-dark">People</a></li>
        
        <li><a href="/html/FAQ.html" class="headClick nav-link px-2 link-dark">FAQs</a></li>
      </ul>
      </div>

      <div class="col-md-3 text-end">
      </div>

    </header>
  </div>

  
  <div class="container">
    <!-- Do not set the `src` attribute -->
    <zero-md>
            <!-- Write your markdown inside a `<script type="text/markdown">` tag -->
            <script type="text/markdown">

# ProcessGT

## Description

Refines and roots gene trees with respect to a rooted binary species tree under MDC criterion. 
The input gene trees must be specified in the Rich Newick Format. Gene trees can 
be either rooted or unrooted. Gene losses are allowed.

The generated output trees will also be generated in the Rich Newick Format. 

Multiple species trees may be specified in which case the tool will run in batch mode with one 
execution of each species tree and set of gene trees.

## Usage

```
Process_GT {species_tree_ident1 [, species_tree_ident2...]} {gene_tree_ident1 [, gene_tree_ident2...]} [-u] 
[-b threshold] [-a taxa map] [result output file]
```

| Settings                                           |                                                                                                                   |           |
|----------------------------------------------------|-------------------------------------------------------------------------------------------------------------------|-----------|
| *{species_tree_ident1 [, species_tree_ident2...]}* | Comma delimited set of species tree identifiers.                                                                  | mandatory |
| *{gene_tree_ident1 [, gene_tree_ident2...]}*       | Comma delimited set of gene tree identifiers.                                                                     | mandatory |
| -u                                                 | Treat gene trees as unrooted.                                                                                     | optional  |
| -b threshold                                       | Specifies bootstrap threshold. Edges in the gene trees that have support lower than threshold will be contracted. | optional  |
| -a taxa map                                        | Gene tree / species tree taxa association.                                                                        | optional  |
| result output file                                 | Optional file destination for command output.                                                                     | optional  |

  

By default, it is assumed that gene trees are rooted. In this case, the method 
refines the gene trees with respect to the species tree. If the gene trees are 
unrooted, users can use option -u. In this case, the method not only refines and 
but also roots gene trees.
  

If input gene trees have bootstrap values a threshold can be set with the -b option.
  

By default, it is assumed that only one individual is sampled per species in gene trees.
However, the option -a allows multiple alleles to be sampled.

## Example


```
#NEXUS
 
BEGIN TREES;
 
Tree speceiesTree = ((((a:1,b:2),c:3),d:4),e:5);
Tree geneTree1 = (((a:1,b:2,c:3),d:4),e:5);
END;
 
 
BEGIN PHYLONET;
 
ProcessGT {speceiesTree} {geneTree1};
 
END;
```



## Command Refernces
* Y. Yu, T. Warnow, and L. Nakhleh. Algorithms for MDC-based multi-locus phylogeny inference. _The 15th Annual International Conference on Research in Computational Molecular Biology (RECOMB)_, pages 531--545, 2011. LNBI 6577.
      </script>
    </zero-md>
</div>
        
                                  
              
    

    <div class="container" height="150px" style="bottom:0;" >
        <footer class="d-flex flex-wrap justify-content-between align-items-center py-3 my-4 border-top">
          <div class="col-md-4 d-flex align-items-center" width="400px">
            <a href="/" class="mb-3 me-2 mb-md-0 text-muted text-decoration-none lh-1">
                <img width="75" height="37.5" src="/media/headerPhoto.png" >
            </a>
            <span class="text-muted" width="350px">&copy; 2022 Luay Nakhleh Lab Group</span>
          </div>
      
          <ul class="nav col-md-4 justify-content-end list-unstyled d-flex">
            <li class="ms-3"><a class="text-muted" href="https://github.com/NakhlehLab"><img width="35" height="35" src="/media/gitLogo.png"></a></li>
            <li class="ms-3"><a class="text-muted" href="https://twitter.com/ricecompsci"><img width="35" height="35" src="/media/twitterLogo.png"></a></li>
          </ul>
        </footer>
     </div>
  
</main>




    <script src="../assets/dist/js/bootstrap.bundle.min.js"></script>

      
  </body>
  
</html>