<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="description" content="">
    <meta name="author" content="Mark Otto, Jacob Thornton, and Bootstrap contributors">
    <meta name="generator" content="Hugo 0.88.1">
    <title>SimBiMarkersinNetwork</title>

    <link rel="canonical" href="https://getbootstrap.com/docs/5.1/examples/headers/">
    <link rel="canonical" href="https://getbootstrap.com/docs/5.1/examples/footers/">
    <link rel="canonical" href="https://getbootstrap.com/docs/5.1/examples/jumbotron/">
    
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bootstrap@4.5.3/dist/css/bootstrap.min.css" integrity="sha384-TX8t27EcRE3e/ihU7zmQxVncDAy5uIKz4rEkgIXeMed4M0jlfIDPvg6uqKI2xXr2" crossorigin="anonymous">


    <!-- Bootstrap core CSS -->
    <link href="../assets/dist/css/bootstrap.min.css" rel="stylesheet">

    <style>
      .bd-placeholder-img {
        font-size: 1.125rem;
        text-anchor: middle;
        -webkit-user-select: none;
        -moz-user-select: none;
        user-select: none;
      }

      @media (min-width: 768px) {
        .bd-placeholder-img-lg {
          font-size: 3.5rem;
        }
      }
    </style>

    
    <!-- Custom styles for this template --> 
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
    <link href="/css/headers.css" rel="stylesheet">
    <link href="/css/footers.css" rel="stylesheet">
    <link href="/css/background.css" rel="stylesheet">
    <link href="/css/tables.css" rel="stylesheet">
    <link href="/css/tutorials.css" rel="stylesheet">
   
  </head>

<body class="htmlBody">
    


<main class="mainContainer">
  <script src="/js/nav.js" charset="utf-8"></script>
  <script src="/js/tutorials.js" charset="utf-8"></script>

  <!-- Lightweight client-side loader that feature-detects and load polyfills only when necessary -->
  <script src="https://cdn.jsdelivr.net/npm/@webcomponents/webcomponentsjs@2/webcomponents-loader.min.js"></script>

  <!-- Load the element definition -->
  <script type="module" src="https://cdn.jsdelivr.net/gh/zerodevx/zero-md@1/src/zero-md.min.js"></script>


  <div class="container" >
    <header class="d-flex flex-wrap align-items-center justify-content-center justify-content-md-between py-3 mb-4 border-bottom">

      <div>
              <a href="/" class="d-flex align-items-center col-md-3 mb-2 mb-md-0 text-dark text-decoration-none">
                <img width="250" height="125" src="/media/headerPhoto.png" >
              </a>
      </div>
      <div>
      <ul class="nav col-12 col-md-auto mb-2 justify-content-center mb-md-0">
        <li><a href="/index.html" class="headClick nav-link px-2 link-secondary">Home</a></li>
        <li><a href="/html/about.html" class="headClick nav-link px-2 link-dark">About</a></li>
        <li><a href="/html/software.html" class="headClick nav-link px-2 link-dark">Software</a></li>
        <li><a href="/html/tutorials.html" class="headClick nav-link px-2 link-dark">Tutorials</a></li>
        <li><a href="https://scholar.google.com/citations?hl=en&user=46HLWf8AAAAJ&view_op=list_works&sortby=pubdate" class="headClick nav-link px-2 link-dark">Publications</a></li>
        <li><a href="/html/people.html" class="headClick nav-link px-2 link-dark">People</a></li>
        
        <li><a href="/html/FAQ.html" class="headClick nav-link px-2 link-dark">FAQs</a></li>
      </ul>
      </div>

      <div class="col-md-3 text-end">
      </div>

    </header>
  </div>

  
  <div class="container">
    <!-- Do not set the `src` attribute -->
    <zero-md>
            <!-- Write your markdown inside a `<script type="text/markdown">` tag -->
            <script type="text/markdown">

# SimBiMarkersinNetwork

## Description

Simulate bi-allelic markers in a phylogenetic network with branch lengths, population mutation rates, and inheritance probabilities.

## Usage

```
SimBiMarkersinNetwork [-pi0 PI0] [-sd seed] [-diploid] [-dominant] -num numMarkers 
[-cu rate] [-tm taxonMap] -truenet trueNetwork [-op]
```

| Settings        |                                                                                                                                                                                                                                                                                                                                                                                                                                                              |           |
|-----------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-----------|
|  *-op*          | Specify whether the generated sequence contains only polymorphic sites.                                                                                                                                                                                                                                                                                                                                                                                      | mandatory |
| *-truenet*      | Specify the true network. If  -cu is not set, the input network should contain divergence times in  units of expected number of mutations per site, inheritance  probabilities and population sizes in units of population mutation rate. If -cu is set, the input network should contain divergence times in units of coalescent units, inheritance probabilities.                                                                                          | mandatory |
|*-num numMarkers*| Specify the number of markers to be generated.                                                                                                                                                                                                                                                                                                                                                                                                               | mandatory |
| -pi0 PI0        | Specify the stationary probability of marker '0'. The default is 0.5.                                                                                                                                                                                                                                                                                                                                                                                        | optional  |
| -sd seed        | The random seed. The default seed is 12345678.                                                                                                                                                                                                                                                                                                                                                                                                               | optional  |
| -diploid        | Specify whether generate sequence from diploids. If this is set without -dominant, the sequence is with 0, 1, 2.                                                                                                                                                                                                                                                                                                                                             | optional  |
| -dominant       | Specify whether generate sequence with dominant markers. If this is set, '0' is dominant.                                                                                                                                                                                                                                                                                                                                                                    | optional  |
| -tm taxonMap    | Gene tree / species tree taxa association.  By default, it is assumed that only one individual is sampled per  species in gene trees. This option allows multiple alleles to be  sampled. For example, the gene tree is (((a1,a2),(b1,b2)),c); and the species tree is ((a,b),c);, the command is -tm <a:a1,a2; b:b1,b2;c:c>. Note that the taxa association should cover all species, e.g. -tm <a:a1,a2; b:b1,b2> is incorrect because c:c is dropped out.  | optional  |
| -cu rate        | Specify whether the true network uses coalescent units. If set, rate will be the overall population mutation rate.                                                                                                                                                                                                                                                                                                                                           | optional  |


## Example


```
#NEXUS

BEGIN PHYLONET;

SimBiMarkersinNetwork -pi0 0.5 -sd 12345678 -num 1000 -cu 0.036 -truenet "(((((Q:0.5)I8#H1 :0.5::0.7,(A:0.5)I6#H2:0.5::0.8)I4:1.0,L:2.0) I3:2.0,(I8#H1:1.0::0.3,R:1.5)I7:2.5)I2:1.0,(( I6#H2:0.5::0.2,C:1.0)I5:1.0,G:2.0)I1:3.0)I0;"
  
END;

```

This command makes Phylonet generate sequence with 1000 sites. The stationary probability 
of marker '0' is set to 0.5. The seed is set to 12345678. The true network contains 
divergence times in units of coalescent units, inheritance probabilities, and 
population mutation rate is set to 0.036.
  
```
#NEXUS

BEGIN PHYLONET;

SimBiMarkersinNetwork -diploid -dominant -op -pi0 0.5 -sd 123456 -num 10000 -tm <A:A_0; B:B_0; C:C_0,C_1,C_2,C_3; D:D_0> -truenet "[0.005]((((C:0.005:0.005)I1#H1:0.006:0.005:0.8,D:0.011:0.005):0.009:0.005,(B:0.014:0.005,I1#H1:0.009:0.005:0.2):0.006:0.005):0.005:0.005,A:0.025:0.005);"

END;
```
  
This command makes Phylonet generate sequence with 10000 sites with only polymorphic sites. 
The bi-allelic markers are from diploid and '0' is the dominant marker. The stationary 
probability of marker '0' is set to 0.5. The seed is set to 12345678. The true network 
contains divergence times in units of expected number of mutations per site, inheritance 
probabilities and population sizes in units of population mutation rate.

  
      </script>
    </zero-md>
</div>
        
                                  
              
    

    <div class="container" height="150px" style="bottom:0;" >
        <footer class="d-flex flex-wrap justify-content-between align-items-center py-3 my-4 border-top">
          <div class="col-md-4 d-flex align-items-center" width="400px">
            <a href="/" class="mb-3 me-2 mb-md-0 text-muted text-decoration-none lh-1">
                <img width="75" height="37.5" src="/media/headerPhoto.png" >
            </a>
            <span class="text-muted" width="350px">&copy; 2022 Luay Nakhleh Lab Group</span>
          </div>
      
          <ul class="nav col-md-4 justify-content-end list-unstyled d-flex">
            <li class="ms-3"><a class="text-muted" href="https://github.com/NakhlehLab"><img width="35" height="35" src="/media/gitLogo.png"></a></li>
            <li class="ms-3"><a class="text-muted" href="https://twitter.com/ricecompsci"><img width="35" height="35" src="/media/twitterLogo.png"></a></li>
          </ul>
        </footer>
     </div>
  
</main>




    <script src="../assets/dist/js/bootstrap.bundle.min.js"></script>

      
  </body>
  
</html>