<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="description" content="">
    <meta name="author" content="Mark Otto, Jacob Thornton, and Bootstrap contributors">
    <meta name="generator" content="Hugo 0.88.1">
    <title>Maximum Likelihood Tree</title>

    <link rel="canonical" href="https://getbootstrap.com/docs/5.1/examples/headers/">
    <link rel="canonical" href="https://getbootstrap.com/docs/5.1/examples/footers/">
    <link rel="canonical" href="https://getbootstrap.com/docs/5.1/examples/jumbotron/">
    
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bootstrap@4.5.3/dist/css/bootstrap.min.css" integrity="sha384-TX8t27EcRE3e/ihU7zmQxVncDAy5uIKz4rEkgIXeMed4M0jlfIDPvg6uqKI2xXr2" crossorigin="anonymous">


    <!-- Bootstrap core CSS -->
    <link href="../assets/dist/css/bootstrap.min.css" rel="stylesheet">

    <style>
      .bd-placeholder-img {
        font-size: 1.125rem;
        text-anchor: middle;
        -webkit-user-select: none;
        -moz-user-select: none;
        user-select: none;
      }

      @media (min-width: 768px) {
        .bd-placeholder-img-lg {
          font-size: 3.5rem;
        }
      }
    </style>

    
    <!-- Custom styles for this template -->
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
    <link href="/css/headers.css" rel="stylesheet">
    <link href="/css/footers.css" rel="stylesheet">
    <link href="/css/background.css" rel="stylesheet">
    <link href="/css/tables.css" rel="stylesheet">
    <link href="/css/tutorials.css" rel="stylesheet">
   
  </head>

<body class="htmlBody">
    


<main class="mainContainer">
  <script src="/js/nav.js" charset="utf-8"></script>
  <script src="/js/tutorials.js" charset="utf-8"></script>
  <script src="https://cdn.jsdelivr.net/gh/google/code-prettify@master/loader/run_prettify.js"></script>


    <div class="container">
        <div class="TutorialBody">
                
                <br>
                <br>
                <h1> ML_TREE</h1>
                <p>
                       Calculating the maximum likelihood tree from a multiple sequence alignment 
                       is easy, and only requires nexus files containing the taxa and sequences.
                </p>
                <p>
                        Currently, there is only support for DNA, but SNP, Bimarkers, and other data types
                        are in the works.
                </p>
                <br>
                

                <pre class="prettyprint">
                        from phylogenpy.tools import ML_TREE
                        from phylogenpy.GTR import *

                        filenames = ["file1.nex", "file2.nex"] 
                        log_output = "out.txt"
                        tree_output = "ml_tree.tree"
                        

                        # Default, with substitution model JC() and number of iterations = 3000
                        likelihoods_1 = ML_TREE(filenames, tree_output, log_output)

                        # num_iter keyword set
                        likelihoods_2 = ML_TREE(filenames, tree_output, log_output, num_iter=5000)

                        # submodel keyword set
                        likelihoods_3 = ML_TREE(filenames, tree_output, log_output, submodel=K2P(.4, .6))
                </pre>

                <br>
                <p>
                        Here is a sample nexus file:
                </p>
                
                <pre class="prettyprint">
                        #NEXUS

                        BEGIN TAXA;

                        DIMENSIONS NTAX=4;

                        TAXLABELS
                        seq_1
                        seq_2
                        seq_3
                        seq_4
                        ;

                        END;

                        BEGIN CHARACTERS;
                        DIMENSIONS NCHAR=5;
                        FORMAT DATATYPE=DNA MISSING=? GAP=- MATCHCHAR=. ;
                        MATRIX

                        seq_1 GACCC

                        seq_2 GACCG

                        seq_3 CACCT

                        seq_4 CACCC
                        ;

                        END;
                </pre>

        </div>
    </div>
      
  
</main>




    <script src="../assets/dist/js/bootstrap.bundle.min.js"></script>

      
  </body>
  
</html>