<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta name="viewport" content="width=device-width, initial-scale=1">
        <meta name="description" content="">
        <meta name="author" content="Mark Kessler">
        <meta name="generator" content="Hugo 0.88.1">
        <title>Botany 2023 Workshop</title>
    
        <link rel="canonical" href="https://getbootstrap.com/docs/5.1/examples/headers/">
        <link rel="canonical" href="https://getbootstrap.com/docs/5.1/examples/footers/">
        <link rel="canonical" href="https://getbootstrap.com/docs/5.1/examples/jumbotron/">
        
        <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bootstrap@4.5.3/dist/css/bootstrap.min.css" integrity="sha384-TX8t27EcRE3e/ihU7zmQxVncDAy5uIKz4rEkgIXeMed4M0jlfIDPvg6uqKI2xXr2" crossorigin="anonymous">
    
    
        <!-- Bootstrap core CSS -->
        <link href="../assets/dist/css/bootstrap.min.css" rel="stylesheet">
    
        <style>
          .bd-placeholder-img {
            font-size: 1.125rem;
            text-anchor: middle;
            -webkit-user-select: none;
            -moz-user-select: none;
            user-select: none;
          }
    
          @media (min-width: 768px) {
            .bd-placeholder-img-lg {
              font-size: 3.5rem;
            }
          }
        </style>
    
        
        <!-- Custom styles for this template -->
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
        <link href="/css/headers.css" rel="stylesheet">
        <link href="/css/footers.css" rel="stylesheet">
        <link href="/css/background.css" rel="stylesheet">
        <link href="/css/tables.css" rel="stylesheet">
        <link href="/css/tutorials.css" rel="stylesheet">
       
      </head>

<body class="htmlBody">
    


<main class="mainContainer">
  <script src="/js/nav.js" charset="utf-8"></script>
  <script src="/js/tutorials.js" charset="utf-8"></script>
  <script src="https://cdn.jsdelivr.net/gh/google/code-prettify@master/loader/run_prettify.js"></script>

  <!-- Lightweight client-side loader that feature-detects and load polyfills only when necessary -->
  <script src="https://cdn.jsdelivr.net/npm/@webcomponents/webcomponentsjs@2/webcomponents-loader.min.js"></script>

  <!-- Load the element definition -->
  <script type="module" src="https://cdn.jsdelivr.net/gh/zerodevx/zero-md@1/src/zero-md.min.js"></script>


  <div class="container" >
    <header class="d-flex flex-wrap align-items-center justify-content-center justify-content-md-between py-3 mb-4 border-bottom">

      <div>
              <a href="/" class="d-flex align-items-center col-md-3 mb-2 mb-md-0 text-dark text-decoration-none">
                <img width="250" height="125" src="/media/headerPhoto.png" >
              </a>
      </div>
      <div>
      <ul class="nav col-12 col-md-auto mb-2 justify-content-center mb-md-0">
        <li><a href="/index.html" class="headClick nav-link px-2 link-secondary">Home</a></li>
        <li><a href="/html/about.html" class="headClick nav-link px-2 link-dark">About</a></li>
        <li><a href="/html/software.html" class="headClick nav-link px-2 link-dark">Software</a></li>
        <li><a href="/html/tutorials.html" class="headClick nav-link px-2 link-dark">Tutorials</a></li>
        <li><a href="https://scholar.google.com/citations?hl=en&user=46HLWf8AAAAJ&view_op=list_works&sortby=pubdate" class="headClick nav-link px-2 link-dark">Publications</a></li>
        <li><a href="/html/people.html" class="headClick nav-link px-2 link-dark">People</a></li>
        
        <li><a href="/html/FAQ.html" class="headClick nav-link px-2 link-dark">FAQs</a></li>
      </ul>
      </div>

      <div class="col-md-3 text-end">
      </div>

    </header>
  </div>

  <div class="container">
    <!-- Do not set the `src` attribute -->
    <zero-md>
            <!-- Write your markdown inside a `<script type="text/markdown">` tag -->
            <script type="text/markdown">

# MP Allop 2.0

## Description
 
Polyploidy has been shown to be prevalent not only in plants but also in other groups of eukaryotic species, but 
most work done thus far on phylogenetic network inference assumes diploid hybridization. 
These inference methods have been applied, with varying degrees of success, to data sets with polyploid species, 
even though polyploidy violates the mathematical assumptions underlying these methods. 
  
In this tutorial, we introduce an improved method (from the previously implemented InferMPAllop in PhyloNet) for inferring parsimonious phylogenetic networks on data that include polyploid species. 
Taking gene tree topologies and a subgenome mapping as input, the method seeks a phylogenetic network that minimizes deep coalescences while accounting for polyploidy.
  
As of July 2023, this method has not yet been optimized for performance/runtime. However, the proposal kernel and its underlying network move
has been improved, such that invalid networks will never be proposed, thereby reducing the potential number of iterations for hill climbing to converge.
Also, with the new kernel, no assumptions about partitioning ploidy taxa need to be made-- every network state can be reached from any other network state
in a certain amount of moves. A proof of this will be provided in an upcoming paper.


## Usage

```
$ python phynetpy.mpallop [gene_trees] [subgenome_map] -i 1000 -s
```

| Settings            |                                                                                                                              |           |
|---------------------|------------------------------------------------------------------------------------------------------------------------------|-----------|
| *gene_trees*        | Comma delimited list of gene tree identifiers.                                                                               | mandatory |
| *subgenome_map*     | Gene tree / subgenome taxa association.                                                                                      | mandatory |
| -i                  | Number of iterations to use for Hill Climbing                                                                                | optional  |
| -s                  | Use Stochastic Hill Climbing instead of classical Hill Climbing                                                              | optional  |

  
The *gene_trees* parameter should be a list of newick strings, separated by commas and surrounded by quotation marks.
  

The *subgenome_map* parameter should be a mapping from network taxa labels to identifiers found in the *gene_trees* newick strings.
  

```
$ python phynetpy.mpallop "((01bA,((01yB,01zB),01xB)),(01aA,(01xA,(01yA,01zA))));" "{"B": ["01bA"], "A": ["01aA"], "X": ["01xA", "01xB"], "Y": ["01yA", "01yB"], "Z": ["01zA", "01zB"]}" -i 1000 -s
```
  

## Installation

PhyNetPy is a python package that is easily installed via PyPI, the Python Package Index.

#### With Pip on Python >= 3.8

Run the following command from a command line prompt:

```
pip install phynetpy
```

#### With an Anaconda VENV

1. Create new environment
  
2. ```conda activate new_env_name```
  
3. ```conda install phynetpy```

  
### Importing in an IDE

First, make sure that IDE you use has the right interpreter of Python enabled.
Visual Studio, for example, shows this information in the bottom right hand corner:
  
![Using Conda Python Interpreter](/media/wconda.png)

  


![Using Local Python Interpreter](/media/wpip.png)

  

Now, we should be ready to code using PhyNetPy!

At the top of your .py file, type
  
```
from PhyNetPy import PhyAllop
```


## Data

For this tutorial, we will be analyzing scenario J from [Zhi Yan et al](https://academic.oup.com/sysbio/article/71/3/706/6380964). The network we are aiming to infer
is the following:


![Scenario J](/media/ScenarioJ.png)

Each highlighted section of the network represents an allopolyploid clade. Any unmarked taxa are non-tetroploid. 

Download ![Scenario J]("/media/nexus/MP_ALLOP_J.txt")
  
  
## Script

```

from PhyNetPy import PhyAllop
from PhyNetPy.GeneTrees import GeneTrees 
from PhyNetPy.NetworkParser import NetworkParser
form PhyNetPy.Graph import DAG
  
dag_list: list[DAG] = NetworkParser("Users/absolute/path/to/MP_ALLOP_J.txt").get_all_networks()
scenarioJ_gene_trees = GeneTrees(dag_list)
  
#Any rng seed will do
# print(scenarioJ_gene_trees.mp_allop_map()) #if you want to view the subgenome mapping for all the taxa in the gene trees
PhyAllop.MP_ALLOP("Users/absolute/path/to/MP_ALLOP_J.txt", scenarioJ_gene_trees.mp_allop_map(), seed = random.randint(0,1000))

```

## Making Sense of the Output


## Command Refernces
* Zhi Yan and others, Maximum Parsimony Inference of Phylogenetic Networks in the Presence of Polyploid Complexes, Systematic Biology, Volume 71, Issue 3, May 2022, Pages 706-720, https://doi.org/10.1093/sysbio/syab081

      </script>
    </zero-md>
</div>
      
    

        
                                  
              
    

    <div class="container" height="150px" style="bottom:0;" >
        <footer class="d-flex flex-wrap justify-content-between align-items-center py-3 my-4 border-top">
          <div class="col-md-4 d-flex align-items-center" width="400px">
            <a href="/" class="mb-3 me-2 mb-md-0 text-muted text-decoration-none lh-1">
                <img width="75" height="37.5" src="/media/headerPhoto.png" >
            </a>
            <span class="text-muted" width="350px">&copy; 2022 Luay Nakhleh Lab Group</span>
          </div>
      
          <ul class="nav col-md-4 justify-content-end list-unstyled d-flex">
            <li class="ms-3"><a class="text-muted" href="https://github.com/NakhlehLab"><img width="35" height="35" src="/media/gitLogo.png"></a></li>
            <li class="ms-3"><a class="text-muted" href="https://twitter.com/ricecompsci"><img width="35" height="35" src="/media/twitterLogo.png"></a></li>
          </ul>
        </footer>
    </div>
  
</main>




    <script src="../assets/dist/js/bootstrap.bundle.min.js"></script>

      
  </body>
  
</html>