<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta name="viewport" content="width=device-width, initial-scale=1">
        <meta name="description" content="">
        <meta name="author" content="Mark Kessler">
        <meta name="generator" content="Hugo 0.88.1">
        <title>Botany 2023 Workshop</title>
    
        <link rel="canonical" href="https://getbootstrap.com/docs/5.1/examples/headers/">
        <link rel="canonical" href="https://getbootstrap.com/docs/5.1/examples/footers/">
        <link rel="canonical" href="https://getbootstrap.com/docs/5.1/examples/jumbotron/">
        
        <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bootstrap@4.5.3/dist/css/bootstrap.min.css" integrity="sha384-TX8t27EcRE3e/ihU7zmQxVncDAy5uIKz4rEkgIXeMed4M0jlfIDPvg6uqKI2xXr2" crossorigin="anonymous">
    
    
        <!-- Bootstrap core CSS -->
        <link href="../assets/dist/css/bootstrap.min.css" rel="stylesheet">
    
        <style>
          .bd-placeholder-img {
            font-size: 1.125rem;
            text-anchor: middle;
            -webkit-user-select: none;
            -moz-user-select: none;
            user-select: none;
          }
    
          @media (min-width: 768px) {
            .bd-placeholder-img-lg {
              font-size: 3.5rem;
            }
          }
        </style>
    
        
        <!-- Custom styles for this template -->
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
        <link href="/css/headers.css" rel="stylesheet">
        <link href="/css/footers.css" rel="stylesheet">
        <link href="/css/background.css" rel="stylesheet">
        <link href="/css/tables.css" rel="stylesheet">
        <link href="/css/tutorials.css" rel="stylesheet">
       
      </head>

<body class="htmlBody">
    


<main class="mainContainer">
  <script src="/js/nav.js" charset="utf-8"></script>
  <script src="/js/tutorials.js" charset="utf-8"></script>
  <script src="https://cdn.jsdelivr.net/gh/google/code-prettify@master/loader/run_prettify.js"></script>

  <!-- Lightweight client-side loader that feature-detects and load polyfills only when necessary -->
  <script src="https://cdn.jsdelivr.net/npm/@webcomponents/webcomponentsjs@2/webcomponents-loader.min.js"></script>

  <!-- Load the element definition -->
  <script type="module" src="https://cdn.jsdelivr.net/gh/zerodevx/zero-md@1/src/zero-md.min.js"></script>


  <div class="container" >
    <header class="d-flex flex-wrap align-items-center justify-content-center justify-content-md-between py-3 mb-4 border-bottom">

      <div>
              <a href="/" class="d-flex align-items-center col-md-3 mb-2 mb-md-0 text-dark text-decoration-none">
                <img width="250" height="125" src="/media/headerPhoto.png" >
              </a>
      </div>
      <div>
      <ul class="nav col-12 col-md-auto mb-2 justify-content-center mb-md-0">
        <li><a href="/index.html" class="headClick nav-link px-2 link-secondary">Home</a></li>
        <li><a href="/html/about.html" class="headClick nav-link px-2 link-dark">About</a></li>
        <li><a href="/html/software.html" class="headClick nav-link px-2 link-dark">Software</a></li>
        <li><a href="/html/tutorials.html" class="headClick nav-link px-2 link-dark">Tutorials</a></li>
        <li><a href="https://scholar.google.com/citations?hl=en&user=46HLWf8AAAAJ&view_op=list_works&sortby=pubdate" class="headClick nav-link px-2 link-dark">Publications</a></li>
        <li><a href="/html/people.html" class="headClick nav-link px-2 link-dark">People</a></li>
        
        <li><a href="/html/FAQ.html" class="headClick nav-link px-2 link-dark">FAQs</a></li>
      </ul>
      </div>

      <div class="col-md-3 text-end">
      </div>

    </header>
  </div>

  <div class="container">
    <!-- Do not set the `src` attribute -->
    <zero-md>
            <!-- Write your markdown inside a `<script type="text/markdown">` tag -->
            <script type="text/markdown">

# MP Allop 2.0

## Description
 
Polyploidy has been shown to be prevalent not only in plants but also in other groups of eukaryotic species, but 
most work done thus far on phylogenetic network inference assumes diploid hybridization. 
These inference methods have been applied, with varying degrees of success, to data sets with polyploid species, 
even though polyploidy violates the mathematical assumptions underlying these methods. 
  
In this tutorial, we introduce an improved method (from the previously implemented InferMPAllop in PhyloNet) for inferring parsimonious phylogenetic networks on data that include polyploid species. 
Taking gene tree topologies and a subgenome mapping as input, the method seeks a phylogenetic network that minimizes deep coalescences while accounting for polyploidy.
  
As of July 2023, this method has not yet been optimized for performance/runtime. However, the proposal kernel and its underlying network move
has been improved, such that invalid networks will never be proposed, thereby reducing the potential number of iterations for hill climbing to converge.
Also, with the new kernel, no assumptions about partitioning ploidy taxa need to be made-- every network state can be reached from any other network state
in a certain amount of moves. A proof of this will be provided in an upcoming paper.


## Usage

At this time, command line usage is not supported. Check the PhyNetPy release page on PyPI for release notes and updates.

```
$ python phynetpy.mpallop [gene_trees] [subgenome_map] -i 1000 -s
```

| Settings            |                                                                                                                              |           |
|---------------------|------------------------------------------------------------------------------------------------------------------------------|-----------|
| *gene_trees*        | Comma delimited list of gene tree identifiers.                                                                               | mandatory |
| *subgenome_map*     | Gene tree / subgenome taxa association.                                                                                      | mandatory |
| -i                  | Number of iterations to use for Hill Climbing                                                                                | optional  |
| -s                  | Use Stochastic Hill Climbing instead of classical Hill Climbing                                                              | optional  |

  
The *gene_trees* parameter should be a list of newick strings, separated by commas and surrounded by quotation marks.
  

The *subgenome_map* parameter should be a mapping from network taxa labels to identifiers found in the *gene_trees* newick strings.
  

```
$ python phynetpy.mpallop "((01bA,((01yB,01zB),01xB)),(01aA,(01xA,(01yA,01zA))));" "{"B": ["01bA"], "A": ["01aA"], "X": ["01xA", "01xB"], "Y": ["01yA", "01yB"], "Z": ["01zA", "01zB"]}" -i 1000 -s
```
  

## Installation

PhyNetPy is a python package that is easily installed via PyPI, the Python Package Index.

#### With Pip on Python >= 3.8

Run the following command from a command line prompt:

```
pip install phynetpy
```

#### With an Anaconda VENV

1. Create new environment
  
2. ```conda activate new_env_name```
  
3. ```conda install phynetpy```

  
### Importing in an IDE

First, make sure that IDE you use has the right interpreter of Python enabled.
Visual Studio, for example, shows this information in the bottom right hand corner:
  
![Using Conda Python Interpreter](/media/wconda.png)
  
![Using Local Python Interpreter](/media/wpip.png)

  
Now, we should be ready to code using PhyNetPy!

At the top of your .py file, type
  
```
from PhyNetPy import PhyAllop
```

Download <a href="/media/nexus/tutorial.py" download>Scenario J Python Script</a>

## Data

For this tutorial, we will be analyzing scenario J from [Zhi Yan et al](https://academic.oup.com/sysbio/article/71/3/706/6380964). The network we are aiming to infer
is the following:


![Scenario J](/media/ScenarioJ.png)
  
Each highlighted section of the network represents an allopolyploid clade. Any unmarked taxa are non-tetroploid. 
  

Download <a href="/media/nexus/J_trimmed.txt" download>Scenario J Trimmed Data</a>  

Download <a href="/media/nexus/J_untrimmed.txt" download>Scenario J Untrimmed Data</a>

#### Subgenome Mappings

The other key piece of information required by MP_ALLOP (other than the list of gene trees) is the subgenome mapping. 
To run analysis on custom datasets, this piece of information needs to be filled out correctly by the user.

Take for example scenario D from [Zhi Yan et al](https://academic.oup.com/sysbio/article/71/3/706/6380964).

The figure displays a Multilabeled Species Tree (MUL tree), with the tetroploid taxa highlighted in green. 
The 8 leaves are labeled as follows: a, b, xA, xB, yA, yB, zA, zB. xA and xB denote 2 copies of gene X, with one found on subgenome A 
and the other on subgenome B. Thus, in scenario D, there are 5 distinct genes which we will capitalize : A, B, X, Y, and Z. Each of X, Y, and Z 
have multiple copies, and thus our mapping for this data in python will be

```python
scenarioD_map = {'A':['a'], 'B': ['b'], 'X': ['xA', 'xB'], 'Y': ['yA', 'yB'], 'Z': ['zA', 'zB']}
```

PhyNetPy offers a method for automating this process of creating a subgenome map. To use it, the user must provide a naming sceme for their
data. PhyNetPy standard naming as a regex is defined as:

```
[0-9]*[a-z][A-Z]
```

The label 01bA (which follows the regex), denotes the 1st sample from gene B on subgenome A. The automated method knows this naming scheme, 
and defaults to it. So if your data uses this system, you can call that method out of the box.

Otherwise, pseudocode for such a naming method may look like this:

```python
def custom_naming_scheme(taxa_label : str) -> str:
    """
    Given a taxa label string, ensure that it 
    1) follows regex rules defined by you
    2) returns the name of the gene to which you'd like to map it to
        ie in PhyNetPy std naming, '01bA' -> 'B'
    """

    key = parse taxa_label for key info
    
    if taxa label is valid name under some regex:
        return key
    else:
        raise Exception("Error parsing taxa_label, does not follow naming convention")

```

Following the definition of a custom naming method:

```python
from PhyNetPy.GeneTrees import GeneTrees
from PhyNetPy.NetworkParser import NetworkParser
from PhyNetPY.Graph import DAG

dag_list: list[DAG] = NetworkParser("Users/absolute/path/to/gene_trees.newick").get_all_networks()

#simply pass the name of the function into the GeneTrees constructor
user_gene_trees : GeneTrees() = GeneTrees(dag_list, naming_rule = custom_naming_scheme)

#to get map
user_gene_trees.mp_allop_map()

```
  
## Scenario J Trimmed Script

```python
import random
from PhyNetPy.PhyAllop import MP_ALLOP
from PhyNetPy.NetworkParser import NetworkParser
from PhyNetPy.GeneTrees import GeneTrees
from PhyNetPy.Graph import DAG

#TODO: replace filename
gene_trees_trimmedJ : list[DAG] = NetworkParser("absolute/path/to/J_trimmed.nex").get_all_networks()

# subgenome mapping for Scenario J
subgenome_map_trimmedJ : dict[str, list[str]]= {'U': ['01uA', '01uB'], 'T': ['01tA', '01tB'], 'B': ['01bA'], 'F': ['01fA'], 
                                                'V': ['01vB', '01vA'], 'C': ['01cA'], 'A': ['01aA'], 'D': ['01dA'], 'O': ['01oA']}

#Using mapping auto generation
#subgenome_map_trimmedJ = GeneTrees(gene_treesJ).mp_allop_map()

output_networks : list[tuple[DAG, int]] = MP_ALLOP(gene_trees_trimmedJ, subgenome_map_trimmedJ)
  
```

## Making Sense of the Output

The MP_ALLOP method by default returns a "top 3" list of (network, parsimony score) tuples, in decreasing order of likelihood (the most likely/end network at index 0). 
Networks can be returned either as newick strings or via the DAG PhyNetPy object. 

```python

output_networks : list[tuple[DAG, int]] = MP_ALLOP(gene_trees_trimmedJ, subgenome_map_trimmedJ)

index = 0
for net, score in output_networks:
    print(f"This network scored {score}!")
   
    #Converts network to a newick string
    print(net.to_newick())
    
    index += 1

```
  

For the trimmed version of scenario J, plugging in the resulting newick string of the most likely network into [IcyTree](https://icytree.org) yields 
  
![Scenario J Trimmed Result](/media/J_trimmed.png)
  
  
Download <a href="/media/nexus/tutorial.py" download>Scenario J Python Script</a>
## Command Refernces
* Zhi Yan and others, Maximum Parsimony Inference of Phylogenetic Networks in the Presence of Polyploid Complexes, Systematic Biology, Volume 71, Issue 3, May 2022, Pages 706-720, https://doi.org/10.1093/sysbio/syab081

      </script>
    </zero-md>
</div>
      
    

        
                                  
              
    

    <div class="container" height="150px" style="bottom:0;" >
        <footer class="d-flex flex-wrap justify-content-between align-items-center py-3 my-4 border-top">
          <div class="col-md-4 d-flex align-items-center" width="400px">
            <a href="/" class="mb-3 me-2 mb-md-0 text-muted text-decoration-none lh-1">
                <img width="75" height="37.5" src="/media/headerPhoto.png" >
            </a>
            <span class="text-muted" width="350px">&copy; 2022 Luay Nakhleh Lab Group</span>
          </div>
      
          <ul class="nav col-md-4 justify-content-end list-unstyled d-flex">
            <li class="ms-3"><a class="text-muted" href="https://github.com/NakhlehLab"><img width="35" height="35" src="/media/gitLogo.png"></a></li>
            <li class="ms-3"><a class="text-muted" href="https://twitter.com/ricecompsci"><img width="35" height="35" src="/media/twitterLogo.png"></a></li>
          </ul>
        </footer>
    </div>
  
</main>




    <script src="../assets/dist/js/bootstrap.bundle.min.js"></script>

      
  </body>
  
</html>