<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="description" content="">
    <meta name="author" content="Mark Otto, Jacob Thornton, and Bootstrap contributors">
    <meta name="generator" content="Hugo 0.88.1">
    <title>Troubleshooting</title>

    <link rel="canonical" href="https://getbootstrap.com/docs/5.1/examples/headers/">
    <link rel="canonical" href="https://getbootstrap.com/docs/5.1/examples/footers/">
    <link rel="canonical" href="https://getbootstrap.com/docs/5.1/examples/jumbotron/">
    
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bootstrap@4.5.3/dist/css/bootstrap.min.css" integrity="sha384-TX8t27EcRE3e/ihU7zmQxVncDAy5uIKz4rEkgIXeMed4M0jlfIDPvg6uqKI2xXr2" crossorigin="anonymous">


    <!-- Bootstrap core CSS -->
    <link href="../assets/dist/css/bootstrap.min.css" rel="stylesheet">

    <style>
      .bd-placeholder-img {
        font-size: 1.125rem;
        text-anchor: middle;
        -webkit-user-select: none;
        -moz-user-select: none;
        user-select: none;
      }

      @media (min-width: 768px) {
        .bd-placeholder-img-lg {
          font-size: 3.5rem;
        }
      }
    </style>

    
    <!-- Custom styles for this template --> 
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
    <link href="/css/headers.css" rel="stylesheet">
    <link href="/css/footers.css" rel="stylesheet">
    <link href="/css/background.css" rel="stylesheet">
    <link href="/css/tables.css" rel="stylesheet">
    <link href="/css/tutorials.css" rel="stylesheet">
   
  </head>

<body class="htmlBody">
    


<main class="mainContainer">
  <script src="/js/nav.js" charset="utf-8"></script>
  <script src="/js/tutorials.js" charset="utf-8"></script>
  <script src="https://cdn.jsdelivr.net/gh/google/code-prettify@master/loader/run_prettify.js"></script>

  <!-- Lightweight client-side loader that feature-detects and load polyfills only when necessary -->
  <script src="https://cdn.jsdelivr.net/npm/@webcomponents/webcomponentsjs@2/webcomponents-loader.min.js"></script>

  <!-- Load the element definition -->
  <script type="module" src="https://cdn.jsdelivr.net/gh/zerodevx/zero-md@1/src/zero-md.min.js"></script>
  <script src="/js/header.js" charset="utf-8"></script>
  <div class="container" >
    <phylogenomics-header></phylogenomics-header>
  </div>

    <div class="container">
        <!-- Do not set the `src` attribute -->
        <zero-md>
                <!-- Write your markdown inside a `<script type="text/markdown">` tag -->
                <script type="text/markdown">
# Troubleshooting Common Problems:

### Outdated Java version
  

``` 
Exception in thread "main" java.lang.UnsupportedClassVersionError: edu/rice/cs/bioinfo/programs/phylonet/Program 
: Unsupported major.minor version 51.0 
```
  

Solution: 
* Update your Java to 1.7.0 or later
  * To check your Java version, type "java -version" on your command line. 
  * To download Java 1.7, please go to website http://www.java.com/en/download/.
  * To link to the new downloaded Java 1.7, for mac, try these two commands from command line: 
    * ```sudo rm /usr/bin/java ```
    * ```sudo ln -s /Library/Internet\ Plug-Ins/JavaAppletPlugin.plugin/Contents/Home/bin/java /usr/bin```



### Wrong format of gene trees

If your gene trees have support values as node labels, PhyloNet cannot read them 
correctly. See Rich Newick Format for the format of phylogenies that PhyloNet reads. 
For example, if your gene tree is ```(((A:1.0,B:1.0)100:1.0,(C:1.0,D:1.0)100:1.0)90:1.0,E:1.0)```, 
PhyloNet will read the support values as the names of nodes so that you will get error indicated below

```
Duplicate node 100
```

Solution: Download the executable <a href="/media/TreeConversion.jar"> TreeConversion.jar </a> and run it on your gene trees. 

TreeConversion runs from command line as follows. It takes two parameters, including 
inputFile which contains your gene trees and outputFile which is the file for the resulting gene trees.
  
``` 
>java -jar PATH_TO_JAR/TreeConversion.jar inputFile outputFile 
```

          
### Not enough memory

Some commands require a large amount of memory, like InferNetwork_ML. If your dataset 
is large, you might run into memory issue indicated below:
  
```
java.lang.OutOfMemoryError: Java heap space
 ```
  

Solution: Increase the memory by using -Xmx.

When you launch PhyloNet from command line, add -Xmx followed by the memory size you want to allocated to Java. 
For example, if you want to increase the memory to 4G for running PhyloNet, run the following command

``` 
>java -jar -Xmx4G PATH/jar/PhyloNet_X.Y.Z.jar script.nex 
```

                </script>
        </zero-md>
    </div>
      
    

        
                                  
              
    

    <div class="container" height="150px" style="bottom:0;" >
        <footer class="d-flex flex-wrap justify-content-between align-items-center py-3 my-4 border-top">
          <div class="col-md-4 d-flex align-items-center" width="400px">
            <a href="/" class="mb-3 me-2 mb-md-0 text-muted text-decoration-none lh-1">
                <img width="75" height="37.5" src="/media/headerPhoto.png" >
            </a>
            <span class="text-muted" width="350px">&copy; 2022 Luay Nakhleh Lab Group</span>
          </div>
      
          <ul class="nav col-md-4 justify-content-end list-unstyled d-flex">
            <li class="ms-3"><a class="text-muted" href="https://github.com/NakhlehLab"><img width="35" height="35" src="/media/gitLogo.png"></a></li>
            <li class="ms-3"><a class="text-muted" href="https://twitter.com/ricecompsci"><img width="35" height="35" src="/media/twitterLogo.png"></a></li>
          </ul>
        </footer>
     </div>
  
</main>




    <script src="../assets/dist/js/bootstrap.bundle.min.js"></script>

      
  </body>
  
</html>