<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="description" content="">
    <meta name="author" content="Mark Kessler">
    <meta name="generator" content="Hugo 0.88.1">
    <title>PhyNetPy Docs</title>

    <link rel="canonical" href="https://getbootstrap.com/docs/5.1/examples/headers/">
    <link rel="canonical" href="https://getbootstrap.com/docs/5.1/examples/footers/">
    <link rel="canonical" href="https://getbootstrap.com/docs/5.1/examples/jumbotron/">
    
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bootstrap@4.5.3/dist/css/bootstrap.min.css" integrity="sha384-TX8t27EcRE3e/ihU7zmQxVncDAy5uIKz4rEkgIXeMed4M0jlfIDPvg6uqKI2xXr2" crossorigin="anonymous">


    <!-- Bootstrap core CSS -->
    <link href="../assets/dist/css/bootstrap.min.css" rel="stylesheet">

    <style>
      .bd-placeholder-img {
        font-size: 1.125rem;
        text-anchor: middle;
        -webkit-user-select: none;
        -moz-user-select: none;
        user-select: none;
      }

      @media (min-width: 768px) {
        .bd-placeholder-img-lg {
          font-size: 3.5rem;
        }
      }
    </style>

    
    <!-- Custom styles for this template --> 
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
    <link href="/css/headers.css" rel="stylesheet">
    <link href="/css/footers.css" rel="stylesheet">
    <link href="/css/background.css" rel="stylesheet">
    <link href="/css/tables.css" rel="stylesheet">
    <link href="/css/tutorials.css" rel="stylesheet">
    <link href="/css/docs.css" rel="stylesheet">
   
  </head>

<body class="htmlBody">
    


<main class="mainContainer">
  <script src="/js/nav.js" charset="utf-8"></script>
  <script src="/js/tutorials.js" charset="utf-8"></script>
  <script src="/js/docs.js" charset="utf-8"></script>
  <script src="https://cdn.jsdelivr.net/gh/google/code-prettify@master/loader/run_prettify.js"></script>

  <!-- Lightweight client-side loader that feature-detects and load polyfills only when necessary -->
  <script src="https://cdn.jsdelivr.net/npm/@webcomponents/webcomponentsjs@2/webcomponents-loader.min.js"></script>

  <!-- Load the element definition -->
  <script type="module" src="https://cdn.jsdelivr.net/gh/zerodevx/zero-md@1/src/zero-md.min.js"></script>

  <!-- Use any element to open the sidenav -->
  

  <div id="main">
        
        <div class="container" >
                <header class="d-flex flex-wrap align-items-center justify-content-center justify-content-md-between py-3 mb-4 border-bottom">
            
                  <div>
                          <a href="/" class="d-flex align-items-center col-md-3 mb-2 mb-md-0 text-dark text-decoration-none">
                            <img width="250" height="125" src="/media/headerPhoto.png" >
                          </a>
                  </div>
                  <div>
                  <ul class="nav col-12 col-md-auto mb-2 justify-content-center mb-md-0">
                    <li><a href="/index.html" class="headClick nav-link px-2 link-secondary">Home</a></li>
                    <li><a href="/html/about.html" class="headClick nav-link px-2 link-dark">About</a></li>
                    <li><a href="/html/software.html" class="headClick nav-link px-2 link-dark">Software</a></li>
                    <li><a href="/html/tutorials.html" class="headClick nav-link px-2 link-dark">Tutorials</a></li>
                    <li><a href="https://scholar.google.com/citations?hl=en&user=46HLWf8AAAAJ&view_op=list_works&sortby=pubdate" class="headClick nav-link px-2 link-dark">Publications</a></li>
                    <li><a href="/html/people.html" class="headClick nav-link px-2 link-dark">People</a></li>
                    
                    <li><a href="/html/FAQ.html" class="headClick nav-link px-2 link-dark">FAQs</a></li>
                  </ul>
                  </div>
            
                  <div class="col-md-3 text-end">
                  </div>
            
                </header>
        </div>

        <div id="mySidenav" class="sidenav">
                <a href="javascript:void(0)" class="closebtn" onclick="closeNav()">&times;</a>
                <a id="link3" class="hoverLink active" onclick="openDoc('net'), linkclick(this)" href="#net">Network</a>
                <a id="link4" class="hoverLink" onclick="openDoc('io'), linkclick(this)" href="#io">IO</a>
                <a id="link5" class="hoverLink" onclick="openDoc('md'), linkclick(this)" href="#md">Method Development</a>
                <a id="link6" class="hoverLink" onclick="openDoc('includes'), linkclick(this)" href="#includes">Included Methods</a>
                <a id="link7" class="hoverLink" onclick="openDoc('extras'), linkclick(this)" href="#extras">Extras</a>
                
        </div>
      
      
        <div class="navArrow">
                <img id="hammy" onclick="openNav()" src="/media/rightArrow.png">
        </div>
      
      <!-- Add all page content inside this div if you want the side nav to push page content to the right (not used if you only want the sidenav to sit on top of the page -->
      
        <div class="container" id="default" style="display:block">
                <zero-md>
                        <script type="text/markdown">
                                # Network Data Structures

* ### PhyNetPy.Network module

  * **NetworkError** class  
  * **NodeError** class  
  * **EdgeError** class  
  * **Node** class  
    * Core building block for phylogenetic networks and population genetics graphs  
  * **Edge** class  
    * Directed edge for connecting two Node objects  
  * **UEdge** class  
    * Undirected edge for connecting two Node objects  
  * **EdgeSet** class  
    * A container/bookkeeper for edges   
  * **NodeSet** class  
    * A container/bookkeeper for nodes  
  * **Network** class  
    * A rooted, directed (Edge objects only), acyclic phylogenetic network  
  * **Graph** class  
    * An unrooted, undirected graph (UEdge objects only).

* ### PhyNetPy.MSA module

  * **MSA** class  
    * Wrapper class for a set of taxa that have one or more DataSequence object associations per taxa.  
  * **DataSequence** class  
    * Wrapper class for a list of biological data symbols (DNA, RNA, Codons, etc.). 

* ### PhyNetPy.Matrix module

  * **MatrixError** class  
  * **Matrix** class

* ### PhyNetPy.GeneTrees module

  * **GeneTreeError** class  
  * **GeneTrees** class  
    * Wrapper class for a set of gene trees (Network objects) defined on a set of taxa

  ### 
                        </script>    
                </zero-md>    
        </div> 
        
        <div class="container" id="net" style="display:none">
                
                <zero-md>
                        <script type="text/markdown">
                                # Network Data Structures

* ### PhyNetPy.Network module

  * **NetworkError** class  
  * **NodeError** class  
  * **EdgeError** class  
  * **Node** class  
    * Core building block for phylogenetic networks and population genetics graphs  
  * **Edge** class  
    * Directed edge for connecting two Node objects  
  * **UEdge** class  
    * Undirected edge for connecting two Node objects  
  * **EdgeSet** class  
    * A container/bookkeeper for edges   
  * **NodeSet** class  
    * A container/bookkeeper for nodes  
  * **Network** class  
    * A rooted, directed (Edge objects only), acyclic phylogenetic network  
  * **Graph** class  
    * An unrooted, undirected graph (UEdge objects only).

* ### PhyNetPy.MSA module

  * **MSA** class  
    * Wrapper class for a set of taxa that have one or more DataSequence object associations per taxa.  
  * **DataSequence** class  
    * Wrapper class for a list of biological data symbols (DNA, RNA, Codons, etc.). 

* ### PhyNetPy.Matrix module

  * **MatrixError** class  
  * **Matrix** class

* ### PhyNetPy.GeneTrees module

  * **GeneTreeError** class  
  * **GeneTrees** class  
    * Wrapper class for a set of gene trees (Network objects) defined on a set of taxa

  ### 
                        </script>    
                </zero-md>    
        </div> 
        
        <div class="container" id="io" style="display:none">
                <zero-md>
                        <script type="text/markdown">
                                # Input/Output

                                * ### PhyNetPy.IO
                                
                                  * NetworkParser class  
                                    * Handles .nex (Nexus) file to Network object conversion  
                                  * NewickParser class  
                                    * Handles .newick (Rich Newick Format) file to .nex/Network object conversion  
                                  * NexusTemplate class  
                                    * Creates .nex file from Network object/MSA data
                        </script>
                </zero-md>
                
        </div>

        <div class="container" id="md" style="display:none">
               
                <zero-md>
                        <script type="text/markdown">
                                # Method Development Tools (Probabilistic Graphical Modeling)

* ### PhyNetPy.ModelGraph

  * **ModelError** class  
  * **Model** class  
    * A Model object is made up of ModelNodes that point to each other and that structurally define what happens when simulations or likelihoods or anything else is called on the Model.  
  * **ModelNode** class  
    * A general class that defines behavior of each node in the Model.  
  * **CalculationNode** class  
    * A type of ModelNode that synthesizes information about the model to make and propagate information about the model.  
  * **StateNode** class  
    * A type of ModelNode that stores information about the Model.  
  * **Accumulator** class  
    * A data storing StateNode that, when a CalculationNode has access to it, can bookkeep information during computations made with respect to the Model.  
  * **Parameter** class  
    * StateNodes that store information about various model parameters for access by the CalculationNodes that use them  
  * **NetworkNode** class  
    * General class that stores network based information in a model  
  * **BranchNode** class  
    * Stores edge information between NetworkNode types  
  * **SubstitionModel** class  
    * CalculationNode that helps other CalculationNodes when they need info about the Q matrix for GTR models  
  * **TaxonSet** class  
    * Holds naming and DataSequence information for each leaf

* ### PhyNetPy.ModelFactory

  * **ModelComponent** class  
    * Defines the general behavior and structure of a custom model component that can be used with the ModelFactory build process  
  * **NetworkComponent** class  
    * Class that is responsible for incorporating a Network object into a Model  
  * **MSAComponent** class  
    * Class that is responsible for incorporating an MSA object into a Model  
      * Dependent on NetworkComponent  
  * **SubstitutionModelComponent** class  
    * Class that is responsible for incorporating a GTR Substitution Model into the Model.  
      * Dependent on NetworkComponent  
  * **ANetworkNode** class  
    * Class that defines general behavior for model network nodes. Implement a subclass to define custom behavior.  
  * **ParameterComponent** class  
    * Class that is responsible for incorporating any parameter into the Model  
  * **ModelFactory** class  
    * Class that takes any number of ModelComponents, and builds them in the proper ordering to create a Model object. 

* ### PhyNetPy.ModelMove

  * **Move** class  
    * Defines general behavior for any custom move that makes changes to the Model object  
  * **AddReticulationMove** class  
    * Adds reticulation edge to a Model’s embedded Network  
  * **RemoveReticulationMove** class  
    * Removes reticulation edge  
  * **FlipReticulationMove** class  
    * Flips the destination and source of a reticulation edge  
  * **RelocateReticulationMove** class  
    * RemoveReticulation \+ AddReticulation in one move  
  * **RelocateReticulationSrcMove** class  
    * Changes the source node of a reticulation edge to a different node  
  * **RelocateReticulationDestMove** class  
    * Changes the destination node of a reticulation edge to a different node  
  * **BranchLength** class  
    * Changes the Branch Length of a given edge  
  * **PSPPMove** class  
    * Custom move for MP-Allop-2.0

* ### PhyNetPy.Search

  * **MetropolisHastings** class  
    * Using a Model and a ProposalKernel with moves that implement a Hastings ratio, run a Metropolis-Hastings search for a given number of iterations.  
  * **HillClimbing** class  
    * Using a Model and a ProposalKernel, run a Hill Climbing search for a given number of iterations  
  * **SimulatedAnnealing** class  
    * Using a Model and a ProposalKernel, run a Simulated Annealing search for a given number of iterations.  
  * **ProposalKernel** class  
    * A general class that implements a generate function that returns a Move that operates on a Model object. Craft a subclass to make custom proposal kernels.
                        </script>
                </zero-md>
                
        </div>

        <div class="container" id="includes" style="display:none">
                <zero-md>
                        <script type="text/markdown">
                                # Included Methods

* ### PhyNetPy.MPAllop

  * **MP\_ALLOP** function  
    * Runs the MPAllop method with a random starting network, given a set of gene trees and a subgenome mapping.  
  * **MP\_ALLOP\_BOOTSTRAP** function  
    * Runs the MPAllop method with a specified starting network, given a set of gene trees and a subgenome mapping.

* ### PhyNetPy.MCMC

  * **MCMC\_SEQ** function  
    * Runs MCMC chain with DNA data from random starting network.  
  * **MCMC\_SEQ\_BOOTSTRAP** function  
    * Runs MCMC chain with DNA data from specified starting network.  
  * **MCMC\_BIMARKERS** function  
    * Runs MCMC chain with SNP data from random starting network.  
  * **MCMC\_BIMARKERS\_BOOTSTRAP** function  
    * Runs MCMC chain with SNP data from specified starting network.
                        </script>
                </zero-md>
                
        </div>

        <div class="container" id="extras" style="display:none">
                <zero-md>
                        <script type="text/markdown">
                                # Phylogenetics Extras

* ### PhyNetPy.Alphabet

  * **Alphabet** class  
    * Provides implementation of various common alphabets for different data that maps characters to numbers, for use in a nexus file or Matrix object. (i.e. the SNP alphabet can be used to map “2002212” to \[2,0,0,2,2,1,2\] or the DNA alphabet can be used to map “AACT” to \[1, 1, 2, 4\]. More details are in the documentation that describe each standard alphabet).

* ### PhyNetPy.Simulations

  * **Yule** class  
    * Simulates random Networks (with no reticulations, so trees) up to a certain time or number of taxa given a birth rate.  
  * **CBDP** class (Constant rate Birth Death Process)  
    * Simulates random Networks (with no reticulations, so trees) up to a certain number of taxa given a birth and death rate.  
  * **LevelK** class  
    * Simulates random Networks via Yule or CBDP but adds reticulations at random and with random inheritance ratios until a level k network is achieved

* ### PhyNetPy.GTR

  * **GTR** class  
    * General Time Reversible substitution model, that serves as a superclass for all the rest of the special case substitution models.  
  * **JC** class  
    * Jukes Cantor ‘69  
  * **HKY** class  
    * Hasegawa–Kishono–Yano ‘85  
  * **K80** class  
    * Kimura ‘80  
  * **SYM** class  
    * Zharkikh ‘94  
  * **TN93** class  
    * Tamura-Nei ‘93  
  * **K81** class  
    * Kimura ‘81  
  * **F81** class  
    * Felsenstein ‘81
                        </script>
                </zero-md> 
        </div>

        
    
        
    <div class="container" height="150px" style="bottom:0;" >
        <footer class="d-flex flex-wrap justify-content-between align-items-center py-3 my-4 border-top">
          <div class="col-md-4 d-flex align-items-center" width="400px">
            <a href="/" class="mb-3 me-2 mb-md-0 text-muted text-decoration-none lh-1">
                <img width="75" height="37.5" src="/media/headerPhoto.png" >
            </a>
            <span class="text-muted" width="350px">&copy; 2022 Luay Nakhleh Lab Group</span>
          </div>
      
          <ul class="nav col-md-4 justify-content-end list-unstyled d-flex">
            <li class="ms-3"><a class="text-muted" href="https://github.com/NakhlehLab"><img width="35" height="35" src="/media/gitLogo.png"></a></li>
            <li class="ms-3"><a class="text-muted" href="https://twitter.com/ricecompsci"><img width="35" height="35" src="/media/twitterLogo.png"></a></li>
          </ul>
        </footer>
     </div>
  </div>
  
</main>




    <script src="../assets/dist/js/bootstrap.bundle.min.js"></script>

      
  </body>
  
</html>