<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="description" content="">
    <meta name="author" content="Mark Otto, Jacob Thornton, and Bootstrap contributors">
    <meta name="generator" content="Hugo 0.88.1">
    <title>Dr. Luay Nakhleh Research Group</title>

    <link rel="canonical" href="https://getbootstrap.com/docs/5.1/examples/headers/">
    <link rel="canonical" href="https://getbootstrap.com/docs/5.1/examples/footers/">
    <link rel="canonical" href="https://getbootstrap.com/docs/5.1/examples/jumbotron/">
    
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bootstrap@4.5.3/dist/css/bootstrap.min.css" integrity="sha384-TX8t27EcRE3e/ihU7zmQxVncDAy5uIKz4rEkgIXeMed4M0jlfIDPvg6uqKI2xXr2" crossorigin="anonymous">


    <!-- Bootstrap core CSS -->
    <link href="../assets/dist/css/bootstrap.min.css" rel="stylesheet">

    <style>
      .bd-placeholder-img {
        font-size: 1.125rem;
        text-anchor: middle;
        -webkit-user-select: none;
        -moz-user-select: none;
        user-select: none;
      }

      @media (min-width: 768px) {
        .bd-placeholder-img-lg {
          font-size: 3.5rem;
        }
      }
    </style>

    
    <!-- Custom styles for this template -->
    <link href="/css/headers.css" rel="stylesheet">
    <link href="/css/footers.css" rel="stylesheet">
    <link href="/css/background.css" rel="stylesheet">
    <link href="/css/tables.css" rel="stylesheet">
    <link href="/css/publications.css" rel="stylesheet">
    <link href="/css/phylo.css" rel="stylesheet">
   
  </head>

<body class="htmlBody">
  <script src="/js/nav.js" charset="utf-8"></script>
      
      <!-- The Modal -->
  <div id="myModal" class="modal">

    <!-- Modal content -->
    <div class="modal-content">
      <span class="close" onclick=closeModal() id="closeBtn">&times;</span>
      <div class="container">
        <a class="btn btn-white btn-animated" href=# download="" id="pdfLink">Download Article PDF</a>
        <a class="btn btn-white btn-animated" href=# download="">Google Scholar Link</a>
        <a class="btn btn-white btn-animated" href=# download="">BibTeX</a>
        <a class="btn btn-white btn-animated" href=# download="">RTF</a>
      </div>
    </div>

  </div>


<main class="mainContainer">
  



  <div class="container">
    <header class="d-flex flex-wrap align-items-center justify-content-center justify-content-md-between py-3 mb-4 border-bottom">

      <a href="/" class="d-flex align-items-center col-md-3 mb-2 mb-md-0 text-dark text-decoration-none">
        <img width="250" height="125" src="/media/headerPhoto.png" >
      </a>

      <ul class="nav col-12 col-md-auto mb-2 justify-content-center mb-md-0">
        <li><a href="/index.html" class="headClick nav-link px-2 link-secondary">Home</a></li>
        <li><a href="/html/pubs.html" class="headClick nav-link px-2 link-dark">Publications</a></li>
        <li><a href="/html/about.html" class="headClick nav-link px-2 link-dark">About</a></li>
        <li><a href="/html/contact.html" class="headClick nav-link px-2 link-dark">Contact</a></li>
        <li><a href="/html/software.html" class="headClick nav-link px-2 link-dark">Software</a></li>
        <li><a href="/html/people.html" class="headClick nav-link px-2 link-dark">People</a></li>
        <li><a href="/html/tutorials.html" class="headClick nav-link px-2 link-dark">Tutorials</a></li>
      </ul>

      <div class="col-md-3 text-end">
      </div>

    </header>
  </div>


    
    <div class="container">
      <div class="centered">
      <table id="slide1">
        <tbody>
          <tr>
            <th class="tableDataBlue spacer"></th>
            <th class="tableDataBlue">
              2020 Publications
            </th>
            <th class="tableDataBlue spacer"></th>
          <tr>
          <tr>
            <td></td>
            <td>Y. Wang, Ogilvie, H. A., and Nakhleh, L., “Practical Speedup of Bayesian Inference of Species Phylogenies by Restricting the Space of Gene Trees”, Molecular Biology and Evolution, vol. 37, pp. 1809-1818, 2020.  
             
            <td>
              <button id="myBtn" onclick=openModal("/media/mark.jpg") >EXPORT</button>
            </td>
          </tr>
          <tr>
            <th class="tableDataBlue"></th>
            <th class="tableDataBlue">
              2019 Publications
            </th>
            <th class="tableDataBlue"></th>
          </tr>
          <tr>
            <td></td>
            <td>R. A. L. Elworth, Ogilvie, H. A., Zhu, J., and Nakhleh, L., “Advances in Computational Methods for Phylogenetic Networks in the Presence of Hybridization”, in Bioinformatics and Phylogenetics: Seminal Contributions of Bernard Moret, vol. 29, Cham: Springer, 2019.
              
            </td>
            <td><a class="exportLink">EXPORT</a></td>
          </tr>
          <tr>
            <td></td>
            <td>M. Edrisi, Zafar, H., and Nakhleh, L., “A Combinatorial Approach for Single-cell Vairant Detection via Phylogenetic Inference”, in Workshop on Algorithms in Bioinformatics (WABI), 2019.
    
            </td>
            <td><a class="exportLink">EXPORT</a></td>
          </tr>
          <tr>
            <td></td>
            <td>Z. Zhu, Liu, X., Ogilvie, H. A., and Nakhleh, L., “A Divide-and-Conquer Method for Scalable Phylogenetic Network Inference from Multi-locus Data”, Bioinformatics (ISMB 19 Proceedings), vol. 35, pp. i370-i378, 2019.
              
            </td>
            <td><a class="exportLink">EXPORT</a></td>
          </tr>
          <tr>
            <td></td>
            <td>G. A. Bravo, Antonelli, A., Bacon, C. D., Bartoszek, K., Blom, M. P. K., Huynh, S., Jones, G., Knowles, L. L., Lamichhaney, S., Marcussen, T., Morlon, H., Nakhleh, L., Oxelman, B., Pfeil, B., Schliep, A., Wahlberg, N., Werneck, F. P., Wiedenhoeft, J., .Willows-Munro, S., and Edwards, S. V., “Embracing heterogeneity: coalescing the Tree of Life and the future of phylogenomics”, PeerJ, vol. 7, p. e6399, 2019.
             
            </td>
            <td><a class="exportLink">EXPORT</a></td>
          </tr>
          <tr>
            <td></td>
            <td>Z. Cao, Zhu, J., and Nakhleh, L., “Empirical performance of tree-based inference of phylogenetic networks”, in Workshop on Algorithms in Bioinformatics (WABI), 2019.
              
            </td>
            <td><a class="exportLink">EXPORT</a></td>
          </tr>
          <tr>
            <td></td>
            <td>Z. Cao, Liu, X., Ogilvie, H. A., Yan, Z., and Nakhleh, L., “Practical Aspects of Phylogenetic Network Analysis Using PhyloNet”, 2019.
              
            </td>
            <td><a class="exportLink">EXPORT</a></td>
          </tr>
          <tr>
            <td></td>
            <td>H. Zafar, Navin, N., Chen, K., and Nakhleh, L., “SiCloneFit: Bayesian inference of population structure, genotype, and phylogeny of tumor clones from single-cell genome sequencing data”, Genome Research, 2019.
              
            </td>
            <td><a class="exportLink">EXPORT</a></td>
          </tr>
          <tr>
            <td></td>
            <td>P. Du, Ogilvie, H. A., and Nakhleh, L., “Unifying Gene Duplication, Loss, and Coalescence on Phylogenetic Networks”, in The International Symposium on Bioinformatics Research and Applications (ISBRA), 2019.

            </td>
            <td><a class="exportLink">EXPORT</a></td>
          </tr>
          <tr>
            <th class="tableDataBlue"></th>
            <th class="tableDataBlue">
              2018 Publications
            </th>
            <th class="tableDataBlue"></th>
          </tr>
          <tr>
            <td></td>
            <td>
              R. A. L. Elworth, Allen, C., Benedict, T., Dulworth, P., and Nakhleh, L., “ALPHA: A toolkit for automated local phylogenomic analyses”, Bioinformatics, vol. 34, no. 16, pp. 2848-2850, 2018.
            </th>
            <td><a class="exportLink">EXPORT</a></td>
          </tr>
          <tr>
            <td></td>
            <td>
              J. Zhu, Wen, D., Yu, Y., Meudt, H., and Nakhleh, L., “Bayesian inference of phylogenetic networks from bi-allelic genetic markers”, PLoS Computational Biology, vol. 14, no. 1, p. e1005932, 2018.
            </th>
            <td><a class="exportLink">EXPORT</a></td>
          </tr>
          <tr>
            <td></td>
            <td>
              D. Wen and Nakhleh, L., “Co-estimating reticulate phylogenies and gene trees from multi-locus sequence data”, Systematic Biology, vol. 67, no. 3, pp. 439-457, 2018.
            </th>
            <td><a class="exportLink">EXPORT</a></td>
          </tr>
          <tr>
            <td></td>
            <td>
              X. Fan, Xu, J., and Nakhleh, L., “Detecting large indels using optical map data”, in RECOMB Comparative Genomics, 2018.
            </th>
            <td><a class="exportLink">EXPORT</a></td>
          </tr>
          <tr>
            <td></td>
            <td>
              R. A. L. Elworth, Allen, C., Benedict, T., Dulworth, P., and Nakhleh, L., “DGEN: A Test Statistic for Detection of General Introgression Scenarios”, Workshop on Algorithms in Bioinformatics (WABI), vol. LIPIcs 19. pp. 19:1 - 19:13, 2018.
            </th>
            <td><a class="exportLink">EXPORT</a></td>
          </tr>
          

          
        </tbody>
      </table>

      <table id="slide2" style="display:none">
        <tbody>
          <tr>
            <th class="tableDataBlue spacer"></th>
            <th class="tableDataBlue">
              2018 Publications
            </th>
            <th class="tableDataBlue spacer"></th>
          <tr>
          <tr>
            <td></td>
            <td>J. Zhu and Nakhleh, L., “Inference of species phylogenies from bi-allelic markers using pseudo-likelihood”, Bioinformatics, vol. 34, pp. i376-385, 2018.             
            <td><a class="exportLink">EXPORT</a></td>
          </tr>
          <tr>
            <td></td>
            <td>D. Wen, Yu, Y., Zhu, J., and Nakhleh, L., “Inferring phylogenetic networks using PhyloNet”, Systematic Biology, vol. 67, no. 4, pp. 735-740, 2018.
            </td>
            <td><a class="exportLink">EXPORT</a></td>
          </tr>
          <tr>
            <td></td>
            <td>B. J. Grindel, Martinez, J. R., Tellman, T. V., Harrington, D. A., Zafar, H., Nakhleh, L., Chung, L., and Farach-Carson, M. C., “Matrilysin/MMP-7 Cleavage of Perlecan/HSPG2 Complexed with Semaphorin 3A Supports FAK-Mediated Stromal Invasion by Prostate Cancer Cells”, Scientific Reports, vol. 8, p. 7262, 2018.
            </td>
            <td><a class="exportLink">EXPORT</a></td>
          </tr>
          <tr>
            <td></td>
            <td>P. Du and Nakhleh, L., “Species tree and reconciliation estimation under a duplication-loss-coalescence model”, The 9th ACM Conference on Bioinformatics, Computational Biology, and Health Informatics (ACM-BCB). pp. 376-385, 2018.             
            </td>
            <td><a class="exportLink">EXPORT</a></td>
          </tr>
          <tr>
            <td></td>
            <td>Y. Wang and Nakhleh, L., “Towards an Accurate and Efficient Heuristic for Species/Gene Tree Co-estimation”, Bioinformatics, vol. 34, pp. i697-i705, 2018.
            </td>
            <td><a class="exportLink">EXPORT</a></td>
          </tr>
          <tr>
            <th class="tableDataBlue spacer"></th>
            <th class="tableDataBlue">
              2017 Publications
            </th>
            <th class="tableDataBlue spacer"></th>
          <tr>
          <tr>
            <td></td>
            <td>
            </td>
            <td><a class="exportLink">EXPORT</a></td>
          </tr>
          <tr>
            <td></td>
            <td>
            </td>
            <td><a class="exportLink">EXPORT</a></td>
          </tr>
          <tr>
            <td></td>
            <td>
            </td>
            <td><a class="exportLink">EXPORT</a></td>
          </tr>
          <tr>
            <td></td>
            <td>
            </td>
            <td><a class="exportLink">EXPORT</a></td>
          </tr>
          <tr>
            <th class="tableDataBlue"></th>
            <th class="tableDataBlue">
              2016 Publications
            </th>
            <th class="tableDataBlue"></th>
          </tr>
        </tbody>
      </table>
      </div>
    </div>

  <!-- TEMPLATE
    <tr>
      <td></td>
      <td>
        Blank
      </th>
      <td><a class="exportLink">EXPORT</a></td>
    </tr>

-->

    <div class="container">
      <div class="centeredPagination">
        <div class="pagination">
          <a onclick=PrevPage() href="#">&laquo;</a>
          <a id="1" onclick=SwitchPage(1) class="active" >1</a>
          <a id="2" onclick=SwitchPage(2)>2</a>
          <a id="3" onclick=SwitchPage(3)>3</a>
          <a id="4" onclick=SwitchPage(4)>4</a>
          <a id="5" onclick=SwitchPage(5)>5</a>
          <a id="6" onclick=SwitchPage(6)>6</a>
          <a onclick=NextPage()>&raquo;</a>
        </div> 
      </div>
    </div>
     
    

        
                                  
              
    

    <div class="container" height="150px" style="bottom:0;" >
        <footer class="d-flex flex-wrap justify-content-between align-items-center py-3 my-4 border-top">
          <div class="col-md-4 d-flex align-items-center" width="400px">
            <a href="/" class="mb-3 me-2 mb-md-0 text-muted text-decoration-none lh-1">
                <img width="75" height="37.5" src="/media/headerPhoto.png" >
            </a>
            <span class="text-muted" width="350px">&copy; 2022 Luay Nakhleh Lab Group</span>
          </div>
      
          <ul class="nav col-md-4 justify-content-end list-unstyled d-flex">
            <li class="ms-3"><a class="text-muted" href="https://github.com/NakhlehLab"><img width="35" height="35" src="/media/gitLogo.png"></a></li>
            <li class="ms-3"><a class="text-muted" href="https://twitter.com/ricecompsci"><img width="35" height="35" src="/media/twitterLogo.png"></a></li>
          </ul>
        </footer>
     </div>
  
</main>




    <script src="../assets/dist/js/bootstrap.bundle.min.js"></script>

      
  </body>
  
</html>