<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="description" content="">
    <meta name="author" content="Mark Otto, Jacob Thornton, and Bootstrap contributors">
    <meta name="generator" content="Hugo 0.88.1">
    <title>PhyloGenPy Sequence Alignments and I/O</title>

    <link rel="canonical" href="https://getbootstrap.com/docs/5.1/examples/headers/">
    <link rel="canonical" href="https://getbootstrap.com/docs/5.1/examples/footers/">
    <link rel="canonical" href="https://getbootstrap.com/docs/5.1/examples/jumbotron/">
    
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bootstrap@4.5.3/dist/css/bootstrap.min.css" integrity="sha384-TX8t27EcRE3e/ihU7zmQxVncDAy5uIKz4rEkgIXeMed4M0jlfIDPvg6uqKI2xXr2" crossorigin="anonymous">


    <!-- Bootstrap core CSS -->
    <link href="../assets/dist/css/bootstrap.min.css" rel="stylesheet">

    <style>
      .bd-placeholder-img {
        font-size: 1.125rem;
        text-anchor: middle;
        -webkit-user-select: none;
        -moz-user-select: none;
        user-select: none;
      }

      @media (min-width: 768px) {
        .bd-placeholder-img-lg {
          font-size: 3.5rem;
        }
      }
    </style>

    
    <!-- Custom styles for this template -->
    <link href="/css/headers.css" rel="stylesheet">
    <link href="/css/footers.css" rel="stylesheet">
    <link href="/css/background.css" rel="stylesheet">
    <link href="/css/tables.css" rel="stylesheet">
    <link href="/css/tutorials.css" rel="stylesheet">
   
  </head>

<body class="htmlBody">
    


<main class="mainContainer">
  <script src="/js/nav.js" charset="utf-8"></script>
  <script src="/js/tutorials.js" charset="utf-8"></script>
  <script src="https://cdn.jsdelivr.net/gh/google/code-prettify@master/loader/run_prettify.js"></script>

  <div class="container">
    <header class="d-flex flex-wrap align-items-center justify-content-center justify-content-md-between py-3 mb-4 border-bottom">

      <a href="/" class="d-flex align-items-center col-md-3 mb-2 mb-md-0 text-dark text-decoration-none">
        <img width="250" height="125" src="/media/headerPhoto.png" >
      </a>

      <ul class="nav col-12 col-md-auto mb-2 justify-content-center mb-md-0">
        <li><a href="/index.html" class="headClick nav-link px-2 link-secondary">Home</a></li>
        <li><a href="/html/pubs.html" class="headClick nav-link px-2 link-dark">Publications</a></li>
        <li><a href="/html/about.html" class="headClick nav-link px-2 link-dark">About</a></li>
        <li><a href="/html/contact.html" class="headClick nav-link px-2 link-dark">Contact</a></li>
        <li><a href="/html/software.html" class="headClick nav-link px-2 link-dark">Software</a></li>
        <li><a href="/html/people.html" class="headClick nav-link px-2 link-dark">People</a></li>
        <li>
          <div class="dropdown-link">
            <a href="/html/tutorials.html" class="headClick nav-link px-2 link-dark">Tutorials</a>
          
            <div class="dropdown-content">
              <h4> PhyloGenPy</h4>
              <a href="/html/phylogenpyDocs.html"> Full Docs </a>
              <a href="/html/treeSim.html"> Tree Simulation </a>
              <a href="/html/networkTutorial.html"> Networks and Nodes </a>
              <a href="/html/treeSim.html"> Bayesian Inference/Metropolis Hastings </a>
              <a href="/html/treeSim.html"> Sequence Data, I/O </a>
              <h4> PhyloNet</h4>
              <a href="/html/gxrf.html"> G-XRF </a>
              <a href="https://wiki.rice.edu/confluence/pages/viewpage.action?pageId=39500205"> Commands And Samples </a>
            </div>
          </div>
        </li>
      </ul>

      <div class="col-md-3 text-end">
      </div>

    </header>
  </div>

    <div class="container">
        <div class="TutorialBody">
                <p>
                        Multiple Sequence Alignments are the building blocks of our Matrix data structure.
                        The Matrix class takes a biopython MSA object and compresses it into a matrix that 
                        contains only the unique site data, while also keeping records to recover the entire matrix.
                        <br>
                        Enabled for the Matrix class is a variety of alphabets, including DNA (the default), RNA, Proteins, Codons, SNP, and Binary.
                </p>
                <p>
                        Currently, there is no reason for a user to explicitly create a Matrix.
                </p>
                <br>
                <p>
                        To create a network is very easy and straightforward:
                </p>

                <pre class="prettyprint">
                        from phylogenpy.Graph import DAG

                        #Can initialize as an empty graph
                        tree = DAG()

                        #To make it a network, simply add in individual nodes
                        #one at a time, or pass in an array of nodes
                        tree.addNodes(individualNodeObj)
                        tree.addNodes(arrayOfNodeObjs)

                        #same applies to adding edges
                        tree.addEdges(individualEdge)
                        tree.addEdges(arrayOfEdges) #Where an edge can just be [NodeObj, NodeObj2]

                        #Now you have a complete network. To grab the newick representation...
                        newickStr = tree.newickString() 

                        #You can now visualize your graph with IcyTree
                        
                        #You can also initialize trees with node and edge lists 
                        tree2 = DAG(edgeList) #add nodes later
                        tree3 = DAG(nodes = nodeList) #add edges later
                        tree4 = DAG(edgeList, nodeList) #add both now


                        #To see a description of all the nodes in your graph:
                        tree.printGraph()

                </pre>

                <br>
                <p>
                        A Node object has the following initialization parameters, all of which are 
                        defaults:
                </p>
                <ol>
                        <li>branchLen = None</li>
                        <li>parNode = [ ]</li>
                        <li>attr = { }</li>
                        <li>isReticulation = False</li>
                        <li>name = None</li>
                </ol>
                <pre class="prettyprint">
                        from phylogenpy.Node import Node

                        #Can initialize as an empty Node
                        new_node = Node()

                        #To set the branch length
                        new_node.setBranchLength(num)
                        
                        #To get the branch length
                        node_len = new_node.branchLen()

                        #To add a parent (must be a Node or UltrametricNode obj)
                        new_node.addParent(par_node)

                        #or to set the parent
                        new_node.setParent(par_node)

                        #To set the name of a Node
                        new_node.setName("Homo Sapiens")

                        #To get the name of the Node
                        node_name = new_node.getName()

                        #To set a custom attribute
                        new_node.setAttribute(attribute, value) #i.e "t" : 5.4
                
                        #To look up an attribute
                        attr_val = new_node.attrLookup(attribute) #i.e "t" returns 5.4

                        #To visualize the values contained in a Node obj 
                        new_node.asString()

                        #A more typical way to create a Node
                        new_node2 = Node(parNode = [par_node], attr = {"t":5.4}, name = "species2")
                        
                </pre>
        </div>
    </div>
      
    

        
                                  
              
    

    <div class="container" height="150px" style="bottom:0;" >
        <footer class="d-flex flex-wrap justify-content-between align-items-center py-3 my-4 border-top">
          <div class="col-md-4 d-flex align-items-center" width="400px">
            <a href="/" class="mb-3 me-2 mb-md-0 text-muted text-decoration-none lh-1">
                <img width="75" height="37.5" src="/media/headerPhoto.png" >
            </a>
            <span class="text-muted" width="350px">&copy; 2022 Luay Nakhleh Lab Group</span>
          </div>
      
          <ul class="nav col-md-4 justify-content-end list-unstyled d-flex">
            <li class="ms-3"><a class="text-muted" href="https://github.com/NakhlehLab"><img width="35" height="35" src="/media/gitLogo.png"></a></li>
            <li class="ms-3"><a class="text-muted" href="https://twitter.com/ricecompsci"><img width="35" height="35" src="/media/twitterLogo.png"></a></li>
          </ul>
        </footer>
     </div>
  
</main>




    <script src="../assets/dist/js/bootstrap.bundle.min.js"></script>

      
  </body>
  
</html>